// fivem-js/lib/hashes/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeatherTypeHash = exports.VehicleWeaponHash = exports.WeaponHash = exports.AmmoType = exports.VehicleHash = exports.PedHash = exports.MaterialHash = void 0;
var MaterialHash_1 = require("./MaterialHash");
Object.defineProperty(exports, "MaterialHash", { enumerable: true, get: function () { return MaterialHash_1.MaterialHash; } });
var PedHash_1 = require("./PedHash");
Object.defineProperty(exports, "PedHash", { enumerable: true, get: function () { return PedHash_1.PedHash; } });
var VehicleHash_1 = require("./VehicleHash");
Object.defineProperty(exports, "VehicleHash", { enumerable: true, get: function () { return VehicleHash_1.VehicleHash; } });
var WeaponHash_1 = require("./WeaponHash");
Object.defineProperty(exports, "AmmoType", { enumerable: true, get: function () { return WeaponHash_1.AmmoType; } });
Object.defineProperty(exports, "WeaponHash", { enumerable: true, get: function () { return WeaponHash_1.WeaponHash; } });
Object.defineProperty(exports, "VehicleWeaponHash", { enumerable: true, get: function () { return WeaponHash_1.VehicleWeaponHash; } });
var WeatherTypeHash_1 = require("./WeatherTypeHash");
Object.defineProperty(exports, "WeatherTypeHash", { enumerable: true, get: function () { return WeatherTypeHash_1.WeatherTypeHash; } });
