// fivem-js/lib/hashes/MaterialHash.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaterialHash = void 0;
var MaterialHash;
(function (MaterialHash) {
    MaterialHash[MaterialHash["None"] = 0] = "None";
    MaterialHash[MaterialHash["Unk"] = 2519482235] = "Unk";
    MaterialHash[MaterialHash["Concrete"] = 1187676648] = "Concrete";
    MaterialHash[MaterialHash["ConcretePothole"] = 359120722] = "ConcretePothole";
    MaterialHash[MaterialHash["ConcreteDusty"] = 3210327185] = "ConcreteDusty";
    MaterialHash[MaterialHash["Tarmac"] = 282940568] = "Tarmac";
    MaterialHash[MaterialHash["TarmacPainted"] = 2993614768] = "TarmacPainted";
    MaterialHash[MaterialHash["TarmacPothole"] = 1886546517] = "TarmacPothole";
    MaterialHash[MaterialHash["RumbleStrip"] = 4044799021] = "RumbleStrip";
    MaterialHash[MaterialHash["BreezeBlock"] = 3340854742] = "BreezeBlock";
    MaterialHash[MaterialHash["Rock"] = 3454750755] = "Rock";
    MaterialHash[MaterialHash["RockMossy"] = 4170197704] = "RockMossy";
    MaterialHash[MaterialHash["Stone"] = 765206029] = "Stone";
    MaterialHash[MaterialHash["Cobblestone"] = 576169331] = "Cobblestone";
    MaterialHash[MaterialHash["Brick"] = 1639053622] = "Brick";
    MaterialHash[MaterialHash["Marble"] = 1945073303] = "Marble";
    MaterialHash[MaterialHash["PavingSlab"] = 1907048430] = "PavingSlab";
    MaterialHash[MaterialHash["SandstoneSolid"] = 592446772] = "SandstoneSolid";
    MaterialHash[MaterialHash["SandstoneBrittle"] = 1913209870] = "SandstoneBrittle";
    MaterialHash[MaterialHash["SandLoose"] = 2699818980] = "SandLoose";
    MaterialHash[MaterialHash["SandCompact"] = 510490462] = "SandCompact";
    MaterialHash[MaterialHash["SandWet"] = 909950165] = "SandWet";
    MaterialHash[MaterialHash["SandTrack"] = 2387446527] = "SandTrack";
    MaterialHash[MaterialHash["SandUnderwater"] = 3158909604] = "SandUnderwater";
    MaterialHash[MaterialHash["SandDryDeep"] = 509508168] = "SandDryDeep";
    MaterialHash[MaterialHash["SandWetDeep"] = 1288448767] = "SandWetDeep";
    MaterialHash[MaterialHash["Ice"] = 3508906581] = "Ice";
    MaterialHash[MaterialHash["IceTarmac"] = 2363942873] = "IceTarmac";
    MaterialHash[MaterialHash["SnowLoose"] = 2357397706] = "SnowLoose";
    MaterialHash[MaterialHash["SnowCompact"] = 3416406407] = "SnowCompact";
    MaterialHash[MaterialHash["SnowDeep"] = 1619704960] = "SnowDeep";
    MaterialHash[MaterialHash["SnowTarmac"] = 1550304810] = "SnowTarmac";
    MaterialHash[MaterialHash["GravelSmall"] = 951832588] = "GravelSmall";
    MaterialHash[MaterialHash["GravelLarge"] = 2128369009] = "GravelLarge";
    MaterialHash[MaterialHash["GravelDeep"] = 3938260814] = "GravelDeep";
    MaterialHash[MaterialHash["GravelTrainTrack"] = 1925605558] = "GravelTrainTrack";
    MaterialHash[MaterialHash["DirtTrack"] = 2409420175] = "DirtTrack";
    MaterialHash[MaterialHash["MudHard"] = 2352068586] = "MudHard";
    MaterialHash[MaterialHash["MudPothole"] = 312396330] = "MudPothole";
    MaterialHash[MaterialHash["MudSoft"] = 1635937914] = "MudSoft";
    MaterialHash[MaterialHash["MudUnderwater"] = 4021477129] = "MudUnderwater";
    MaterialHash[MaterialHash["MudDeep"] = 1109728704] = "MudDeep";
    MaterialHash[MaterialHash["Marsh"] = 223086562] = "Marsh";
    MaterialHash[MaterialHash["MarshDeep"] = 1584636462] = "MarshDeep";
    MaterialHash[MaterialHash["Soil"] = 3594309083] = "Soil";
    MaterialHash[MaterialHash["ClayHard"] = 1144315879] = "ClayHard";
    MaterialHash[MaterialHash["ClaySoft"] = 560985072] = "ClaySoft";
    MaterialHash[MaterialHash["GrassLong"] = 3833216577] = "GrassLong";
    MaterialHash[MaterialHash["Grass"] = 1333033863] = "Grass";
    MaterialHash[MaterialHash["GrassShort"] = 3008270349] = "GrassShort";
    MaterialHash[MaterialHash["Hay"] = 2461440131] = "Hay";
    MaterialHash[MaterialHash["Bushes"] = 581794674] = "Bushes";
    MaterialHash[MaterialHash["Twigs"] = 3381615457] = "Twigs";
    MaterialHash[MaterialHash["Leaves"] = 2253637325] = "Leaves";
    MaterialHash[MaterialHash["Woodchips"] = 3985845843] = "Woodchips";
    MaterialHash[MaterialHash["TreeBark"] = 2379541433] = "TreeBark";
    MaterialHash[MaterialHash["MetalSolidSmall"] = 2847687191] = "MetalSolidSmall";
    MaterialHash[MaterialHash["MetalSolidMedium"] = 3929336056] = "MetalSolidMedium";
    MaterialHash[MaterialHash["MetalSolidLarge"] = 752131025] = "MetalSolidLarge";
    MaterialHash[MaterialHash["MetalHollowSmall"] = 15972667] = "MetalHollowSmall";
    MaterialHash[MaterialHash["MetalHollowMedium"] = 1849540536] = "MetalHollowMedium";
    MaterialHash[MaterialHash["MetalHollowLarge"] = 3711753465] = "MetalHollowLarge";
    MaterialHash[MaterialHash["MetalChainlinkSmall"] = 762193613] = "MetalChainlinkSmall";
    MaterialHash[MaterialHash["MetalChainlinkLarge"] = 125958708] = "MetalChainlinkLarge";
    MaterialHash[MaterialHash["MetalCorrugatedIron"] = 834144982] = "MetalCorrugatedIron";
    MaterialHash[MaterialHash["MetalGrille"] = 3868849285] = "MetalGrille";
    MaterialHash[MaterialHash["MetalRailing"] = 2100727187] = "MetalRailing";
    MaterialHash[MaterialHash["MetalDuct"] = 1761524221] = "MetalDuct";
    MaterialHash[MaterialHash["MetalGarageDoor"] = 4063706601] = "MetalGarageDoor";
    MaterialHash[MaterialHash["MetalManhole"] = 3539969597] = "MetalManhole";
    MaterialHash[MaterialHash["WoodSolidSmall"] = 3895095068] = "WoodSolidSmall";
    MaterialHash[MaterialHash["WoodSolidMedium"] = 555004797] = "WoodSolidMedium";
    MaterialHash[MaterialHash["WoodSolidLarge"] = 815762359] = "WoodSolidLarge";
    MaterialHash[MaterialHash["WoodSolidPolished"] = 126470059] = "WoodSolidPolished";
    MaterialHash[MaterialHash["WoodFloorDusty"] = 3545514974] = "WoodFloorDusty";
    MaterialHash[MaterialHash["WoodHollowSmall"] = 1993976879] = "WoodHollowSmall";
    MaterialHash[MaterialHash["WoodHollowMedium"] = 3929491133] = "WoodHollowMedium";
    MaterialHash[MaterialHash["WoodHollowLarge"] = 3369548007] = "WoodHollowLarge";
    MaterialHash[MaterialHash["WoodChipboard"] = 1176309403] = "WoodChipboard";
    MaterialHash[MaterialHash["WoodOldCreaky"] = 722686013] = "WoodOldCreaky";
    MaterialHash[MaterialHash["WoodHighDensity"] = 2552123904] = "WoodHighDensity";
    MaterialHash[MaterialHash["WoodLattice"] = 2011204130] = "WoodLattice";
    MaterialHash[MaterialHash["Ceramic"] = 3108646581] = "Ceramic";
    MaterialHash[MaterialHash["RoofTile"] = 1755188853] = "RoofTile";
    MaterialHash[MaterialHash["RoofFelt"] = 2877802565] = "RoofFelt";
    MaterialHash[MaterialHash["Fibreglass"] = 1354180827] = "Fibreglass";
    MaterialHash[MaterialHash["Tarpaulin"] = 3652308448] = "Tarpaulin";
    MaterialHash[MaterialHash["Plastic"] = 2221655295] = "Plastic";
    MaterialHash[MaterialHash["PlasticHollow"] = 627123000] = "PlasticHollow";
    MaterialHash[MaterialHash["PlasticHighDensity"] = 2668971817] = "PlasticHighDensity";
    MaterialHash[MaterialHash["PlasticClear"] = 2435246283] = "PlasticClear";
    MaterialHash[MaterialHash["PlasticHollowClear"] = 772722531] = "PlasticHollowClear";
    MaterialHash[MaterialHash["PlasticHighDensityClear"] = 2956494126] = "PlasticHighDensityClear";
    MaterialHash[MaterialHash["FibreglassHollow"] = 3528912198] = "FibreglassHollow";
    MaterialHash[MaterialHash["Rubber"] = 4149231379] = "Rubber";
    MaterialHash[MaterialHash["RubberHollow"] = 3511032624] = "RubberHollow";
    MaterialHash[MaterialHash["Linoleum"] = 289630530] = "Linoleum";
    MaterialHash[MaterialHash["Laminate"] = 1845676458] = "Laminate";
    MaterialHash[MaterialHash["CarpetSolid"] = 669292054] = "CarpetSolid";
    MaterialHash[MaterialHash["CarpetSolidDusty"] = 158576196] = "CarpetSolidDusty";
    MaterialHash[MaterialHash["CarpetFloorboard"] = 2898482353] = "CarpetFloorboard";
    MaterialHash[MaterialHash["Cloth"] = 122789469] = "Cloth";
    MaterialHash[MaterialHash["PlasterSolid"] = 3720844863] = "PlasterSolid";
    MaterialHash[MaterialHash["PlasterBrittle"] = 4043078398] = "PlasterBrittle";
    MaterialHash[MaterialHash["CardboardSheet"] = 236511221] = "CardboardSheet";
    MaterialHash[MaterialHash["CardboardBox"] = 2885912856] = "CardboardBox";
    MaterialHash[MaterialHash["Paper"] = 474149820] = "Paper";
    MaterialHash[MaterialHash["Foam"] = 808719444] = "Foam";
    MaterialHash[MaterialHash["FeatherPillow"] = 1341866303] = "FeatherPillow";
    MaterialHash[MaterialHash["Polystyrene"] = 2538039965] = "Polystyrene";
    MaterialHash[MaterialHash["Leather"] = 3724496396] = "Leather";
    MaterialHash[MaterialHash["Tvscreen"] = 1429989756] = "Tvscreen";
    MaterialHash[MaterialHash["SlattedBlinds"] = 673696729] = "SlattedBlinds";
    MaterialHash[MaterialHash["GlassShootThrough"] = 937503243] = "GlassShootThrough";
    MaterialHash[MaterialHash["GlassBulletproof"] = 244521486] = "GlassBulletproof";
    MaterialHash[MaterialHash["GlassOpaque"] = 1500272081] = "GlassOpaque";
    MaterialHash[MaterialHash["Perspex"] = 2675173228] = "Perspex";
    MaterialHash[MaterialHash["CarMetal"] = 4201905313] = "CarMetal";
    MaterialHash[MaterialHash["CarPlastic"] = 2137197282] = "CarPlastic";
    MaterialHash[MaterialHash["CarSofttop"] = 3315319434] = "CarSofttop";
    MaterialHash[MaterialHash["CarSofttopClear"] = 2130571536] = "CarSofttopClear";
    MaterialHash[MaterialHash["CarGlassWeak"] = 1247281098] = "CarGlassWeak";
    MaterialHash[MaterialHash["CarGlassMedium"] = 602884284] = "CarGlassMedium";
    MaterialHash[MaterialHash["CarGlassStrong"] = 1070994698] = "CarGlassStrong";
    MaterialHash[MaterialHash["CarGlassBulletproof"] = 2573051366] = "CarGlassBulletproof";
    MaterialHash[MaterialHash["CarGlassOpaque"] = 513061559] = "CarGlassOpaque";
    MaterialHash[MaterialHash["Water"] = 435688960] = "Water";
    MaterialHash[MaterialHash["Blood"] = 5236042] = "Blood";
    MaterialHash[MaterialHash["Oil"] = 3660485991] = "Oil";
    MaterialHash[MaterialHash["Petrol"] = 2660782956] = "Petrol";
    MaterialHash[MaterialHash["FreshMeat"] = 868733839] = "FreshMeat";
    MaterialHash[MaterialHash["DriedMeat"] = 2849806867] = "DriedMeat";
    MaterialHash[MaterialHash["EmissiveGlass"] = 1501078253] = "EmissiveGlass";
    MaterialHash[MaterialHash["EmissivePlastic"] = 1059629996] = "EmissivePlastic";
    MaterialHash[MaterialHash["VfxMetalElectrified"] = 3985833031] = "VfxMetalElectrified";
    MaterialHash[MaterialHash["VfxMetalWaterTower"] = 611561919] = "VfxMetalWaterTower";
    MaterialHash[MaterialHash["VfxMetalSteam"] = 3603690002] = "VfxMetalSteam";
    MaterialHash[MaterialHash["VfxMetalFlame"] = 332778253] = "VfxMetalFlame";
    MaterialHash[MaterialHash["PhysNoFriction"] = 1666473731] = "PhysNoFriction";
    MaterialHash[MaterialHash["PhysGolfBall"] = 2601153738] = "PhysGolfBall";
    MaterialHash[MaterialHash["PhysTennisBall"] = 4038262533] = "PhysTennisBall";
    MaterialHash[MaterialHash["PhysCaster"] = 4059664613] = "PhysCaster";
    MaterialHash[MaterialHash["PhysCasterRusty"] = 2016463089] = "PhysCasterRusty";
    MaterialHash[MaterialHash["PhysCarVoid"] = 1345867677] = "PhysCarVoid";
    MaterialHash[MaterialHash["PhysPedCapsule"] = 4003336261] = "PhysPedCapsule";
    MaterialHash[MaterialHash["PhysElectricFence"] = 3124923563] = "PhysElectricFence";
    MaterialHash[MaterialHash["PhysElectricMetal"] = 2281206151] = "PhysElectricMetal";
    MaterialHash[MaterialHash["PhysBarbedWire"] = 2751643840] = "PhysBarbedWire";
    MaterialHash[MaterialHash["PhysPooltableSurface"] = 605776921] = "PhysPooltableSurface";
    MaterialHash[MaterialHash["PhysPooltableCushion"] = 972939963] = "PhysPooltableCushion";
    MaterialHash[MaterialHash["PhysPooltableBall"] = 3546625734] = "PhysPooltableBall";
    MaterialHash[MaterialHash["Buttocks"] = 483400232] = "Buttocks";
    MaterialHash[MaterialHash["ThighLeft"] = 3834431425] = "ThighLeft";
    MaterialHash[MaterialHash["ShinLeft"] = 652772852] = "ShinLeft";
    MaterialHash[MaterialHash["FootLeft"] = 1926285543] = "FootLeft";
    MaterialHash[MaterialHash["ThighRight"] = 4057986041] = "ThighRight";
    MaterialHash[MaterialHash["ShinRight"] = 3848931141] = "ShinRight";
    MaterialHash[MaterialHash["FootRight"] = 2925830612] = "FootRight";
    MaterialHash[MaterialHash["Spine0"] = 2372680412] = "Spine0";
    MaterialHash[MaterialHash["Spine1"] = 3154854427] = "Spine1";
    MaterialHash[MaterialHash["Spine2"] = 1457572381] = "Spine2";
    MaterialHash[MaterialHash["Spine3"] = 32752644] = "Spine3";
    MaterialHash[MaterialHash["ClavicleLeft"] = 2825350831] = "ClavicleLeft";
    MaterialHash[MaterialHash["UpperArmLeft"] = 3784624938] = "UpperArmLeft";
    MaterialHash[MaterialHash["LowerArmLeft"] = 1045062756] = "LowerArmLeft";
    MaterialHash[MaterialHash["HandLeft"] = 113101985] = "HandLeft";
    MaterialHash[MaterialHash["ClavicleRight"] = 2737678298] = "ClavicleRight";
    MaterialHash[MaterialHash["UpperArmRight"] = 1501153539] = "UpperArmRight";
    MaterialHash[MaterialHash["LowerArmRight"] = 1777921590] = "LowerArmRight";
    MaterialHash[MaterialHash["HandRight"] = 2000961972] = "HandRight";
    MaterialHash[MaterialHash["Neck"] = 1718294164] = "Neck";
    MaterialHash[MaterialHash["Head"] = 3559574543] = "Head";
    MaterialHash[MaterialHash["AnimalDefault"] = 286224918] = "AnimalDefault";
    MaterialHash[MaterialHash["CarEngine"] = 2378027672] = "CarEngine";
    MaterialHash[MaterialHash["Puddle"] = 999829011] = "Puddle";
    MaterialHash[MaterialHash["ConcretePavement"] = 2015599386] = "ConcretePavement";
    MaterialHash[MaterialHash["BrickPavement"] = 3147605720] = "BrickPavement";
    MaterialHash[MaterialHash["PhysDynamicCoverBound"] = 2247498441] = "PhysDynamicCoverBound";
    MaterialHash[MaterialHash["VfxWoodBeerBarrel"] = 998201806] = "VfxWoodBeerBarrel";
    MaterialHash[MaterialHash["WoodHighFriction"] = 2154880249] = "WoodHighFriction";
    MaterialHash[MaterialHash["RockNoinst"] = 127813971] = "RockNoinst";
    MaterialHash[MaterialHash["BushesNoinst"] = 1441114862] = "BushesNoinst";
    MaterialHash[MaterialHash["MetalSolidRoadSurface"] = 3565854962] = "MetalSolidRoadSurface";
    MaterialHash[MaterialHash["StuntRampSurface"] = 2206792300] = "StuntRampSurface";
    MaterialHash[MaterialHash["Temp01"] = 746881105] = "Temp01";
    MaterialHash[MaterialHash["Temp02"] = 2316997185] = "Temp02";
    MaterialHash[MaterialHash["Temp03"] = 1911121241] = "Temp03";
    MaterialHash[MaterialHash["Temp04"] = 1923995104] = "Temp04";
    MaterialHash[MaterialHash["Temp05"] = 2901304848] = "Temp05";
    MaterialHash[MaterialHash["Temp06"] = 1061250033] = "Temp06";
    MaterialHash[MaterialHash["Temp07"] = 2529443614] = "Temp07";
    MaterialHash[MaterialHash["Temp08"] = 1343679702] = "Temp08";
    MaterialHash[MaterialHash["Temp09"] = 1026054937] = "Temp09";
    MaterialHash[MaterialHash["Temp10"] = 63305994] = "Temp10";
    MaterialHash[MaterialHash["Temp11"] = 47470226] = "Temp11";
    MaterialHash[MaterialHash["Temp12"] = 702596674] = "Temp12";
    MaterialHash[MaterialHash["Temp13"] = 2657481383] = "Temp13";
    MaterialHash[MaterialHash["Temp14"] = 3649011722] = "Temp14";
    MaterialHash[MaterialHash["Temp15"] = 2710969365] = "Temp15";
    MaterialHash[MaterialHash["Temp16"] = 2782232023] = "Temp16";
    MaterialHash[MaterialHash["Temp17"] = 1011960114] = "Temp17";
    MaterialHash[MaterialHash["Temp18"] = 1354993138] = "Temp18";
    MaterialHash[MaterialHash["Temp19"] = 3493162850] = "Temp19";
    MaterialHash[MaterialHash["Temp20"] = 2242086891] = "Temp20";
    MaterialHash[MaterialHash["Temp21"] = 3257211236] = "Temp21";
    MaterialHash[MaterialHash["Temp22"] = 3674578943] = "Temp22";
    MaterialHash[MaterialHash["Temp23"] = 465002639] = "Temp23";
    MaterialHash[MaterialHash["Temp24"] = 1963820161] = "Temp24";
    MaterialHash[MaterialHash["Temp25"] = 1952288305] = "Temp25";
    MaterialHash[MaterialHash["Temp26"] = 3178714198] = "Temp26";
    MaterialHash[MaterialHash["Temp27"] = 889255498] = "Temp27";
    MaterialHash[MaterialHash["Temp28"] = 3115293198] = "Temp28";
    MaterialHash[MaterialHash["Temp29"] = 1078418101] = "Temp29";
    MaterialHash[MaterialHash["Temp30"] = 13626292] = "Temp30";
})(MaterialHash = exports.MaterialHash || (exports.MaterialHash = {}));
