// fivem-js/lib/hashes/PedHash.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PedHash = void 0;
var PedHash;
(function (PedHash) {
    PedHash[PedHash["Michael"] = 225514697] = "Michael";
    PedHash[PedHash["Franklin"] = 2602752943] = "Franklin";
    PedHash[PedHash["Trevor"] = 2608926626] = "Trevor";
    PedHash[PedHash["Abigail"] = 1074457665] = "Abigail";
    PedHash[PedHash["Agent"] = 3614493108] = "Agent";
    PedHash[PedHash["Agent14"] = 4227433577] = "Agent14";
    PedHash[PedHash["AmandaTownley"] = 1830688247] = "AmandaTownley";
    PedHash[PedHash["Andreas"] = 1206185632] = "Andreas";
    PedHash[PedHash["Ashley"] = 2129936603] = "Ashley";
    PedHash[PedHash["AviSchwartzman"] = 939183526] = "AviSchwartzman";
    PedHash[PedHash["Ballasog"] = 2802535058] = "Ballasog";
    PedHash[PedHash["Bankman"] = 2426248831] = "Bankman";
    PedHash[PedHash["Barry"] = 797459875] = "Barry";
    PedHash[PedHash["Bestmen"] = 1464257942] = "Bestmen";
    PedHash[PedHash["Beverly"] = 3181518428] = "Beverly";
    PedHash[PedHash["Brad"] = 3183167778] = "Brad";
    PedHash[PedHash["Bride"] = 1633872967] = "Bride";
    PedHash[PedHash["Car3Guy1"] = 2230970679] = "Car3Guy1";
    PedHash[PedHash["Car3Guy2"] = 1975732938] = "Car3Guy2";
    PedHash[PedHash["Casey"] = 3774489940] = "Casey";
    PedHash[PedHash["Chef"] = 1240128502] = "Chef";
    PedHash[PedHash["Chef2"] = 2240322243] = "Chef2";
    PedHash[PedHash["Clay"] = 1825562762] = "Clay";
    PedHash[PedHash["Claypain"] = 2634057640] = "Claypain";
    PedHash[PedHash["Cletus"] = 3865252245] = "Cletus";
    PedHash[PedHash["CrisFormage"] = 678319271] = "CrisFormage";
    PedHash[PedHash["Dale"] = 1182012905] = "Dale";
    PedHash[PedHash["DaveNorton"] = 365775923] = "DaveNorton";
    PedHash[PedHash["Denise"] = 2181772221] = "Denise";
    PedHash[PedHash["Devin"] = 1952555184] = "Devin";
    PedHash[PedHash["DoaMan"] = 1646160893] = "DoaMan";
    PedHash[PedHash["Dom"] = 2620240008] = "Dom";
    PedHash[PedHash["Dreyfuss"] = 3666413874] = "Dreyfuss";
    PedHash[PedHash["DrFriedlander"] = 3422293493] = "DrFriedlander";
    PedHash[PedHash["EdToh"] = 712602007] = "EdToh";
    PedHash[PedHash["Fabien"] = 3499148112] = "Fabien";
    PedHash[PedHash["FbiSuit01"] = 988062523] = "FbiSuit01";
    PedHash[PedHash["Floyd"] = 2981205682] = "Floyd";
    PedHash[PedHash["G"] = 2216405299] = "G";
    PedHash[PedHash["Groom"] = 4274948997] = "Groom";
    PedHash[PedHash["Hao"] = 1704428387] = "Hao";
    PedHash[PedHash["Hunter"] = 3457361118] = "Hunter";
    PedHash[PedHash["Janet"] = 225287241] = "Janet";
    PedHash[PedHash["JayNorris"] = 2050158196] = "JayNorris";
    PedHash[PedHash["Jewelass"] = 257763003] = "Jewelass";
    PedHash[PedHash["JimmyBoston"] = 3986688045] = "JimmyBoston";
    PedHash[PedHash["JimmyDisanto"] = 1459905209] = "JimmyDisanto";
    PedHash[PedHash["JoeMinuteman"] = 3189787803] = "JoeMinuteman";
    PedHash[PedHash["JohnnyKlebitz"] = 2278195374] = "JohnnyKlebitz";
    PedHash[PedHash["Josef"] = 3776618420] = "Josef";
    PedHash[PedHash["Josh"] = 2040438510] = "Josh";
    PedHash[PedHash["KarenDaniels"] = 3948009817] = "KarenDaniels";
    PedHash[PedHash["KerryMcintosh"] = 1530648845] = "KerryMcintosh";
    PedHash[PedHash["LamarDavis"] = 1706635382] = "LamarDavis";
    PedHash[PedHash["Lazlow"] = 3756278757] = "Lazlow";
    PedHash[PedHash["LesterCrest"] = 1302784073] = "LesterCrest";
    PedHash[PedHash["Lifeinvad01"] = 1401530684] = "Lifeinvad01";
    PedHash[PedHash["Lifeinvad02"] = 666718676] = "Lifeinvad02";
    PedHash[PedHash["Magenta"] = 4242313482] = "Magenta";
    PedHash[PedHash["Malc"] = 4055673113] = "Malc";
    PedHash[PedHash["Manuel"] = 4248931856] = "Manuel";
    PedHash[PedHash["Marnie"] = 411185872] = "Marnie";
    PedHash[PedHash["MaryAnn"] = 2741999622] = "MaryAnn";
    PedHash[PedHash["Maude"] = 1005070462] = "Maude";
    PedHash[PedHash["Michelle"] = 3214308084] = "Michelle";
    PedHash[PedHash["Milton"] = 3408943538] = "Milton";
    PedHash[PedHash["Molly"] = 2936266209] = "Molly";
    PedHash[PedHash["MrK"] = 3990661997] = "MrK";
    PedHash[PedHash["MrsPhillips"] = 946007720] = "MrsPhillips";
    PedHash[PedHash["MrsThornhill"] = 503621995] = "MrsThornhill";
    PedHash[PedHash["Natalia"] = 3726105915] = "Natalia";
    PedHash[PedHash["NervousRon"] = 3170921201] = "NervousRon";
    PedHash[PedHash["Nigel"] = 3367442045] = "Nigel";
    PedHash[PedHash["OldMan1a"] = 1906124788] = "OldMan1a";
    PedHash[PedHash["OldMan2"] = 4011150407] = "OldMan2";
    PedHash[PedHash["Omega"] = 1625728984] = "Omega";
    PedHash[PedHash["ONeil"] = 768005095] = "ONeil";
    PedHash[PedHash["Orleans"] = 1641334641] = "Orleans";
    PedHash[PedHash["Ortega"] = 648372919] = "Ortega";
    PedHash[PedHash["Paper"] = 2577072326] = "Paper";
    PedHash[PedHash["Patricia"] = 3312325004] = "Patricia";
    PedHash[PedHash["Popov"] = 645279998] = "Popov";
    PedHash[PedHash["Paige"] = 357551935] = "Paige";
    PedHash[PedHash["Priest"] = 1681385341] = "Priest";
    PedHash[PedHash["PrologueDriver"] = 2237544099] = "PrologueDriver";
    PedHash[PedHash["PrologueSec01"] = 1888624839] = "PrologueSec01";
    PedHash[PedHash["PrologueSec02"] = 666086773] = "PrologueSec02";
    PedHash[PedHash["RampGang"] = 3845001836] = "RampGang";
    PedHash[PedHash["RampHic"] = 1165307954] = "RampHic";
    PedHash[PedHash["RampHipster"] = 3740245870] = "RampHipster";
    PedHash[PedHash["RampMex"] = 3870061732] = "RampMex";
    PedHash[PedHash["Rashkovsky"] = 940326374] = "Rashkovsky";
    PedHash[PedHash["RoccoPelosi"] = 3585757951] = "RoccoPelosi";
    PedHash[PedHash["RussianDrunk"] = 1024089777] = "RussianDrunk";
    PedHash[PedHash["ScreenWriter"] = 4293277303] = "ScreenWriter";
    PedHash[PedHash["SiemonYetarian"] = 1283141381] = "SiemonYetarian";
    PedHash[PedHash["Solomon"] = 2260598310] = "Solomon";
    PedHash[PedHash["SteveHains"] = 941695432] = "SteveHains";
    PedHash[PedHash["Stretch"] = 915948376] = "Stretch";
    PedHash[PedHash["Talina"] = 3885222120] = "Talina";
    PedHash[PedHash["Tanisha"] = 226559113] = "Tanisha";
    PedHash[PedHash["TaoCheng"] = 3697041061] = "TaoCheng";
    PedHash[PedHash["TaosTranslator"] = 2089096292] = "TaosTranslator";
    PedHash[PedHash["TennisCoach"] = 2721800023] = "TennisCoach";
    PedHash[PedHash["Terry"] = 1728056212] = "Terry";
    PedHash[PedHash["TomEpsilon"] = 3447159466] = "TomEpsilon";
    PedHash[PedHash["Tonya"] = 3402126148] = "Tonya";
    PedHash[PedHash["TracyDisanto"] = 3728026165] = "TracyDisanto";
    PedHash[PedHash["TrafficWarden"] = 1461287021] = "TrafficWarden";
    PedHash[PedHash["TylerDixon"] = 1382414087] = "TylerDixon";
    PedHash[PedHash["VagosSpeak"] = 4194109068] = "VagosSpeak";
    PedHash[PedHash["Wade"] = 2459507570] = "Wade";
    PedHash[PedHash["WeiCheng"] = 2867128955] = "WeiCheng";
    PedHash[PedHash["Zimbor"] = 188012277] = "Zimbor";
    PedHash[PedHash["AbigailCutscene"] = 2306246977] = "AbigailCutscene";
    PedHash[PedHash["AgentCutscene"] = 3614493108] = "AgentCutscene";
    PedHash[PedHash["Agent14Cutscene"] = 1841036427] = "Agent14Cutscene";
    PedHash[PedHash["AmandaTownleyCutscene"] = 2515474659] = "AmandaTownleyCutscene";
    PedHash[PedHash["AndreasCutscene"] = 3881194279] = "AndreasCutscene";
    PedHash[PedHash["AnitaCutscene"] = 117698822] = "AnitaCutscene";
    PedHash[PedHash["AntonCutscene"] = 2781317046] = "AntonCutscene";
    PedHash[PedHash["AshleyCutscene"] = 650367097] = "AshleyCutscene";
    PedHash[PedHash["AviSchwartzmanCutscene"] = 2560490906] = "AviSchwartzmanCutscene";
    PedHash[PedHash["BallasogCutscene"] = 2884567044] = "BallasogCutscene";
    PedHash[PedHash["BankmanCutscene"] = 2539657518] = "BankmanCutscene";
    PedHash[PedHash["BarryCutscene"] = 1767447799] = "BarryCutscene";
    PedHash[PedHash["BeverlyCutscene"] = 3027157846] = "BeverlyCutscene";
    PedHash[PedHash["BradCutscene"] = 4024807398] = "BradCutscene";
    PedHash[PedHash["BradCadaverCutscene"] = 1915268960] = "BradCadaverCutscene";
    PedHash[PedHash["BrideCutscene"] = 2193587873] = "BrideCutscene";
    PedHash[PedHash["BurgerDrugCutscene"] = 2363277399] = "BurgerDrugCutscene";
    PedHash[PedHash["Car3Guy1Cutscene"] = 71501447] = "Car3Guy1Cutscene";
    PedHash[PedHash["Car3Guy2Cutscene"] = 327394568] = "Car3Guy2Cutscene";
    PedHash[PedHash["CarBuyerCutscene"] = 2362341647] = "CarBuyerCutscene";
    PedHash[PedHash["CaseyCutscene"] = 3935738944] = "CaseyCutscene";
    PedHash[PedHash["ChefCutscene"] = 2739391114] = "ChefCutscene";
    PedHash[PedHash["Chef2Cutscene"] = 2925257274] = "Chef2Cutscene";
    PedHash[PedHash["ChinGoonCutscene"] = 2831296918] = "ChinGoonCutscene";
    PedHash[PedHash["ClayCutscene"] = 3687553076] = "ClayCutscene";
    PedHash[PedHash["CletusCutscene"] = 3404326357] = "CletusCutscene";
    PedHash[PedHash["CopCutscene"] = 2595446627] = "CopCutscene";
    PedHash[PedHash["CrisFormageCutscene"] = 3253960934] = "CrisFormageCutscene";
    PedHash[PedHash["CustomerCutscene"] = 2756669323] = "CustomerCutscene";
    PedHash[PedHash["DaleCutscene"] = 216536661] = "DaleCutscene";
    PedHash[PedHash["DaveNortonCutscene"] = 2240226444] = "DaveNortonCutscene";
    PedHash[PedHash["DebraCutscene"] = 3973074921] = "DebraCutscene";
    PedHash[PedHash["DeniseCutscene"] = 1870669624] = "DeniseCutscene";
    PedHash[PedHash["DeniseFriendCutscene"] = 3045926185] = "DeniseFriendCutscene";
    PedHash[PedHash["DevinCutscene"] = 788622594] = "DevinCutscene";
    PedHash[PedHash["DomCutscene"] = 1198698306] = "DomCutscene";
    PedHash[PedHash["DreyfussCutscene"] = 1012965715] = "DreyfussCutscene";
    PedHash[PedHash["DrFriedlanderCutscene"] = 2745392175] = "DrFriedlanderCutscene";
    PedHash[PedHash["FabienCutscene"] = 1191403201] = "FabienCutscene";
    PedHash[PedHash["FbiSuit01Cutscene"] = 1482427218] = "FbiSuit01Cutscene";
    PedHash[PedHash["FloydCutscene"] = 103106535] = "FloydCutscene";
    PedHash[PedHash["FosRepCutscene"] = 466359675] = "FosRepCutscene";
    PedHash[PedHash["GCutscene"] = 2727244247] = "GCutscene";
    PedHash[PedHash["GroomCutscene"] = 2058033618] = "GroomCutscene";
    PedHash[PedHash["GroveStrDlrCutscene"] = 3898166818] = "GroveStrDlrCutscene";
    PedHash[PedHash["GuadalopeCutscene"] = 261428209] = "GuadalopeCutscene";
    PedHash[PedHash["GurkCutscene"] = 3272931111] = "GurkCutscene";
    PedHash[PedHash["HaoCutscene"] = 3969814300] = "HaoCutscene";
    PedHash[PedHash["HughCutscene"] = 1863555924] = "HughCutscene";
    PedHash[PedHash["HunterCutscene"] = 1531218220] = "HunterCutscene";
    PedHash[PedHash["ImranCutscene"] = 3812756443] = "ImranCutscene";
    PedHash[PedHash["JackHowitzerCutscene"] = 1153203121] = "JackHowitzerCutscene";
    PedHash[PedHash["JanetCutscene"] = 808778210] = "JanetCutscene";
    PedHash[PedHash["JanitorCutscene"] = 3254803008] = "JanitorCutscene";
    PedHash[PedHash["JewelassCutscene"] = 1145088004] = "JewelassCutscene";
    PedHash[PedHash["JimmyBostonCutscene"] = 60192701] = "JimmyBostonCutscene";
    PedHash[PedHash["JimmyDisantoCutscene"] = 3100414644] = "JimmyDisantoCutscene";
    PedHash[PedHash["JoeMinutemanCutscene"] = 4036845097] = "JoeMinutemanCutscene";
    PedHash[PedHash["JohnnyKlebitzCutscene"] = 4203395201] = "JohnnyKlebitzCutscene";
    PedHash[PedHash["JosefCutscene"] = 1167549130] = "JosefCutscene";
    PedHash[PedHash["JoshCutscene"] = 1158606749] = "JoshCutscene";
    PedHash[PedHash["KarenDanielsCutscene"] = 1269774364] = "KarenDanielsCutscene";
    PedHash[PedHash["LamarDavisCutscene"] = 1162230285] = "LamarDavisCutscene";
    PedHash[PedHash["LazlowCutscene"] = 949295643] = "LazlowCutscene";
    PedHash[PedHash["LesterCrestCutscene"] = 3046438339] = "LesterCrestCutscene";
    PedHash[PedHash["Lifeinvad01Cutscene"] = 1918178165] = "Lifeinvad01Cutscene";
    PedHash[PedHash["MagentaCutscene"] = 1477887514] = "MagentaCutscene";
    PedHash[PedHash["ManuelCutscene"] = 4222842058] = "ManuelCutscene";
    PedHash[PedHash["MarnieCutscene"] = 1464721716] = "MarnieCutscene";
    PedHash[PedHash["MartinMadrazoCutscene"] = 1129928304] = "MartinMadrazoCutscene";
    PedHash[PedHash["MaryannCutscene"] = 161007533] = "MaryannCutscene";
    PedHash[PedHash["MaudeCutscene"] = 3166991819] = "MaudeCutscene";
    PedHash[PedHash["MerryWeatherCutscene"] = 1631478380] = "MerryWeatherCutscene";
    PedHash[PedHash["MichelleCutscene"] = 1890499016] = "MichelleCutscene";
    PedHash[PedHash["MiltonCutscene"] = 3077190415] = "MiltonCutscene";
    PedHash[PedHash["MollyCutscene"] = 1167167044] = "MollyCutscene";
    PedHash[PedHash["MoviePremFemaleCutscene"] = 1270514905] = "MoviePremFemaleCutscene";
    PedHash[PedHash["MoviePremMaleCutscene"] = 2372398717] = "MoviePremMaleCutscene";
    PedHash[PedHash["MrKCutscene"] = 3284966005] = "MrKCutscene";
    PedHash[PedHash["MrsPhillipsCutscene"] = 3422397391] = "MrsPhillipsCutscene";
    PedHash[PedHash["MrsThornhillCutscene"] = 1334976110] = "MrsThornhillCutscene";
    PedHash[PedHash["NataliaCutscene"] = 1325314544] = "NataliaCutscene";
    PedHash[PedHash["NervousRonCutscene"] = 2023152276] = "NervousRonCutscene";
    PedHash[PedHash["NigelCutscene"] = 3779566603] = "NigelCutscene";
    PedHash[PedHash["OldMan1aCutscene"] = 518814684] = "OldMan1aCutscene";
    PedHash[PedHash["OldMan2Cutscene"] = 2566514544] = "OldMan2Cutscene";
    PedHash[PedHash["OmegaCutscene"] = 2339419141] = "OmegaCutscene";
    PedHash[PedHash["OrleansCutscene"] = 2905870170] = "OrleansCutscene";
    PedHash[PedHash["OrtegaCutscene"] = 3235579087] = "OrtegaCutscene";
    PedHash[PedHash["OscarCutscene"] = 4095687067] = "OscarCutscene";
    PedHash[PedHash["PaigeCutscene"] = 1528799427] = "PaigeCutscene";
    PedHash[PedHash["PaperCutscene"] = 1798879480] = "PaperCutscene";
    PedHash[PedHash["PopovCutscene"] = 1635617250] = "PopovCutscene";
    PedHash[PedHash["PatriciaCutscene"] = 3750433537] = "PatriciaCutscene";
    PedHash[PedHash["PornDudesCutscene"] = 793443893] = "PornDudesCutscene";
    PedHash[PedHash["PriestCutscene"] = 1299047806] = "PriestCutscene";
    PedHash[PedHash["PrologueDriverCutscene"] = 4027271643] = "PrologueDriverCutscene";
    PedHash[PedHash["PrologueSec01Cutscene"] = 2141384740] = "PrologueSec01Cutscene";
    PedHash[PedHash["PrologueSec02Cutscene"] = 512955554] = "PrologueSec02Cutscene";
    PedHash[PedHash["RampGangCutscene"] = 3263172030] = "RampGangCutscene";
    PedHash[PedHash["RampHicCutscene"] = 2240582840] = "RampHicCutscene";
    PedHash[PedHash["RampHipsterCutscene"] = 569740212] = "RampHipsterCutscene";
    PedHash[PedHash["RampMarineCutscene"] = 1634506681] = "RampMarineCutscene";
    PedHash[PedHash["RampMexCutscene"] = 4132362192] = "RampMexCutscene";
    PedHash[PedHash["RashkovskyCutscene"] = 411081129] = "RashkovskyCutscene";
    PedHash[PedHash["ReporterCutscene"] = 776079908] = "ReporterCutscene";
    PedHash[PedHash["RoccoPelosiCutscene"] = 2858686092] = "RoccoPelosiCutscene";
    PedHash[PedHash["RussianDrunkCutscene"] = 1179785778] = "RussianDrunkCutscene";
    PedHash[PedHash["ScreenWriterCutscene"] = 2346790124] = "ScreenWriterCutscene";
    PedHash[PedHash["SiemonYetarianCutscene"] = 3230888450] = "SiemonYetarianCutscene";
    PedHash[PedHash["SolomonCutscene"] = 4140949582] = "SolomonCutscene";
    PedHash[PedHash["SteveHainsCutscene"] = 2766184958] = "SteveHainsCutscene";
    PedHash[PedHash["StretchCutscene"] = 2302502917] = "StretchCutscene";
    PedHash[PedHash["Stripper01Cutscene"] = 2934601397] = "Stripper01Cutscene";
    PedHash[PedHash["Stripper02Cutscene"] = 2168724337] = "Stripper02Cutscene";
    PedHash[PedHash["TanishaCutscene"] = 1123963760] = "TanishaCutscene";
    PedHash[PedHash["TaoChengCutscene"] = 2288257085] = "TaoChengCutscene";
    PedHash[PedHash["TaosTranslatorCutscene"] = 1397974313] = "TaosTranslatorCutscene";
    PedHash[PedHash["TennisCoachCutscene"] = 1545995274] = "TennisCoachCutscene";
    PedHash[PedHash["TerryCutscene"] = 978452933] = "TerryCutscene";
    PedHash[PedHash["TomCutscene"] = 1776856003] = "TomCutscene";
    PedHash[PedHash["TomEpsilonCutscene"] = 2349847778] = "TomEpsilonCutscene";
    PedHash[PedHash["TonyaCutscene"] = 1665391897] = "TonyaCutscene";
    PedHash[PedHash["TracyDisantoCutscene"] = 101298480] = "TracyDisantoCutscene";
    PedHash[PedHash["TrafficWardenCutscene"] = 3727243251] = "TrafficWardenCutscene";
    PedHash[PedHash["UndercoverCopCutscene"] = 4017642090] = "UndercoverCopCutscene";
    PedHash[PedHash["VagosSpeakCutscene"] = 1224690857] = "VagosSpeakCutscene";
    PedHash[PedHash["WadeCutscene"] = 3529955798] = "WadeCutscene";
    PedHash[PedHash["WeiChengCutscene"] = 819699067] = "WeiChengCutscene";
    PedHash[PedHash["ZimborCutscene"] = 3937184496] = "ZimborCutscene";
    PedHash[PedHash["Boar"] = 3462393972] = "Boar";
    PedHash[PedHash["Cat"] = 1462895032] = "Cat";
    PedHash[PedHash["ChickenHawk"] = 2864127842] = "ChickenHawk";
    PedHash[PedHash["Chimp"] = 2825402133] = "Chimp";
    PedHash[PedHash["Chop"] = 351016938] = "Chop";
    PedHash[PedHash["Cormorant"] = 1457690978] = "Cormorant";
    PedHash[PedHash["Cow"] = 4244282910] = "Cow";
    PedHash[PedHash["Coyote"] = 1682622302] = "Coyote";
    PedHash[PedHash["Crow"] = 402729631] = "Crow";
    PedHash[PedHash["Deer"] = 3630914197] = "Deer";
    PedHash[PedHash["Dolphin"] = 2344268885] = "Dolphin";
    PedHash[PedHash["Fish"] = 802685111] = "Fish";
    PedHash[PedHash["Hen"] = 1794449327] = "Hen";
    PedHash[PedHash["HammerShark"] = 1015224100] = "HammerShark";
    PedHash[PedHash["Humpback"] = 1193010354] = "Humpback";
    PedHash[PedHash["Husky"] = 1318032802] = "Husky";
    PedHash[PedHash["KillerWhale"] = 2374682809] = "KillerWhale";
    PedHash[PedHash["MountainLion"] = 307287994] = "MountainLion";
    PedHash[PedHash["Pig"] = 2971380566] = "Pig";
    PedHash[PedHash["Pigeon"] = 111281960] = "Pigeon";
    PedHash[PedHash["Poodle"] = 1125994524] = "Poodle";
    PedHash[PedHash["Pug"] = 1832265812] = "Pug";
    PedHash[PedHash["Rabbit"] = 3753204865] = "Rabbit";
    PedHash[PedHash["Rat"] = 3283429734] = "Rat";
    PedHash[PedHash["Retriever"] = 882848737] = "Retriever";
    PedHash[PedHash["Rhesus"] = 3268439891] = "Rhesus";
    PedHash[PedHash["Rottweiler"] = 2506301981] = "Rottweiler";
    PedHash[PedHash["Seagull"] = 3549666813] = "Seagull";
    PedHash[PedHash["Shepherd"] = 1126154828] = "Shepherd";
    PedHash[PedHash["Stingray"] = 2705875277] = "Stingray";
    PedHash[PedHash["TigerShark"] = 113504370] = "TigerShark";
    PedHash[PedHash["Westy"] = 2910340283] = "Westy";
    PedHash[PedHash["Abner"] = 4037813798] = "Abner";
    PedHash[PedHash["AlDiNapoli"] = 4042020578] = "AlDiNapoli";
    PedHash[PedHash["Antonb"] = 3479321132] = "Antonb";
    PedHash[PedHash["Armoured01"] = 3455013896] = "Armoured01";
    PedHash[PedHash["Babyd"] = 3658575486] = "Babyd";
    PedHash[PedHash["Bankman01"] = 3272005365] = "Bankman01";
    PedHash[PedHash["Baygor"] = 1380197501] = "Baygor";
    PedHash[PedHash["Benny"] = 3300333010] = "Benny";
    PedHash[PedHash["BikeHire01"] = 1984382277] = "BikeHire01";
    PedHash[PedHash["BikerChic"] = 4198014287] = "BikerChic";
    PedHash[PedHash["BoatStaff01M"] = 3361671816] = "BoatStaff01M";
    PedHash[PedHash["BoatStaff01F"] = 848542878] = "BoatStaff01F";
    PedHash[PedHash["BurgerDrug"] = 2340239206] = "BurgerDrug";
    PedHash[PedHash["Chip"] = 610290475] = "Chip";
    PedHash[PedHash["Claude01"] = 3237179831] = "Claude01";
    PedHash[PedHash["ClubHouseBar01"] = 1914945105] = "ClubHouseBar01";
    PedHash[PedHash["CocaineFemale01"] = 1897303236] = "CocaineFemale01";
    PedHash[PedHash["CocaineMale01"] = 3455927962] = "CocaineMale01";
    PedHash[PedHash["ComJane"] = 3064628686] = "ComJane";
    PedHash[PedHash["Corpse01"] = 773063444] = "Corpse01";
    PedHash[PedHash["Corpse02"] = 228356856] = "Corpse02";
    PedHash[PedHash["CounterfeitFemale01"] = 1074385436] = "CounterfeitFemale01";
    PedHash[PedHash["CounterfeitMale01"] = 2625926338] = "CounterfeitMale01";
    PedHash[PedHash["Cyclist01"] = 755956971] = "Cyclist01";
    PedHash[PedHash["DeadHooker"] = 1943971979] = "DeadHooker";
    PedHash[PedHash["Drowned"] = 1943971979] = "Drowned";
    PedHash[PedHash["ExArmy01"] = 1161072059] = "ExArmy01";
    PedHash[PedHash["ExecutivePAMale01"] = 983887149] = "ExecutivePAMale01";
    PedHash[PedHash["ExecutivePAFemale01"] = 2913175640] = "ExecutivePAFemale01";
    PedHash[PedHash["Famdd01"] = 866411749] = "Famdd01";
    PedHash[PedHash["FibArchitect"] = 874722259] = "FibArchitect";
    PedHash[PedHash["FibMugger01"] = 2243544680] = "FibMugger01";
    PedHash[PedHash["FibSec01"] = 1558115333] = "FibSec01";
    PedHash[PedHash["FilmDirector"] = 728636342] = "FilmDirector";
    PedHash[PedHash["FilmNoir"] = 732742363] = "FilmNoir";
    PedHash[PedHash["Finguru01"] = 1189322339] = "Finguru01";
    PedHash[PedHash["ForgeryFemale01"] = 3691903615] = "ForgeryFemale01";
    PedHash[PedHash["ForgeryMale01"] = 325317957] = "ForgeryMale01";
    PedHash[PedHash["FreemodeFemale01"] = 2627665880] = "FreemodeFemale01";
    PedHash[PedHash["FreemodeMale01"] = 1885233650] = "FreemodeMale01";
    PedHash[PedHash["Glenstank01"] = 1169888870] = "Glenstank01";
    PedHash[PedHash["Griff01"] = 3293887675] = "Griff01";
    PedHash[PedHash["Guido01"] = 3333724719] = "Guido01";
    PedHash[PedHash["GunVend01"] = 3005388626] = "GunVend01";
    PedHash[PedHash["Hacker"] = 2579169528] = "Hacker";
    PedHash[PedHash["HeliStaff01"] = 431423238] = "HeliStaff01";
    PedHash[PedHash["Hippie01"] = 4030826507] = "Hippie01";
    PedHash[PedHash["Hotposh01"] = 2526768638] = "Hotposh01";
    PedHash[PedHash["Imporage"] = 880829941] = "Imporage";
    PedHash[PedHash["Jesus01"] = 3459037009] = "Jesus01";
    PedHash[PedHash["Jewelass01"] = 4040474158] = "Jewelass01";
    PedHash[PedHash["JewelSec01"] = 2899099062] = "JewelSec01";
    PedHash[PedHash["JewelThief"] = 3872144604] = "JewelThief";
    PedHash[PedHash["Justin"] = 2109968527] = "Justin";
    PedHash[PedHash["Mani"] = 3367706194] = "Mani";
    PedHash[PedHash["Markfost"] = 479578891] = "Markfost";
    PedHash[PedHash["Marston01"] = 943915367] = "Marston01";
    PedHash[PedHash["MethFemale01"] = 3778572496] = "MethFemale01";
    PedHash[PedHash["MethMale01"] = 1293671805] = "MethMale01";
    PedHash[PedHash["MilitaryBum"] = 1191548746] = "MilitaryBum";
    PedHash[PedHash["Miranda"] = 1095737979] = "Miranda";
    PedHash[PedHash["Mistress"] = 1573528872] = "Mistress";
    PedHash[PedHash["Misty01"] = 3509125021] = "Misty01";
    PedHash[PedHash["MovieStar"] = 894928436] = "MovieStar";
    PedHash[PedHash["MPros01"] = 1822283721] = "MPros01";
    PedHash[PedHash["Niko01"] = 4007317449] = "Niko01";
    PedHash[PedHash["Paparazzi"] = 1346941736] = "Paparazzi";
    PedHash[PedHash["Party01"] = 921110016] = "Party01";
    PedHash[PedHash["PartyTarget"] = 2180468199] = "PartyTarget";
    PedHash[PedHash["PestContDriver"] = 994527967] = "PestContDriver";
    PedHash[PedHash["PestContGunman"] = 193469166] = "PestContGunman";
    PedHash[PedHash["Pogo01"] = 3696858125] = "Pogo01";
    PedHash[PedHash["Poppymich"] = 602513566] = "Poppymich";
    PedHash[PedHash["Princess"] = 3538133636] = "Princess";
    PedHash[PedHash["Prisoner01"] = 2073775040] = "Prisoner01";
    PedHash[PedHash["PrologueHostage01"] = 3306347811] = "PrologueHostage01";
    PedHash[PedHash["PrologueMournFemale01"] = 2718472679] = "PrologueMournFemale01";
    PedHash[PedHash["PrologueMournMale01"] = 3465937675] = "PrologueMournMale01";
    PedHash[PedHash["RivalPaparazzi"] = 1624626906] = "RivalPaparazzi";
    PedHash[PedHash["ShopKeep01"] = 416176080] = "ShopKeep01";
    PedHash[PedHash["SpyActor"] = 2886641112] = "SpyActor";
    PedHash[PedHash["SpyActress"] = 1535236204] = "SpyActress";
    PedHash[PedHash["StripperLite"] = 695248020] = "StripperLite";
    PedHash[PedHash["Taphillbilly"] = 2585681490] = "Taphillbilly";
    PedHash[PedHash["Tramp01"] = 1787764635] = "Tramp01";
    PedHash[PedHash["VagosFun01"] = 3299219389] = "VagosFun01";
    PedHash[PedHash["WillyFist"] = 2423691919] = "WillyFist";
    PedHash[PedHash["WeedFemale01"] = 1596374223] = "WeedFemale01";
    PedHash[PedHash["WeedMale01"] = 2648833641] = "WeedMale01";
    PedHash[PedHash["Zombie01"] = 2890614022] = "Zombie01";
    PedHash[PedHash["Acult01AMM"] = 1413662315] = "Acult01AMM";
    PedHash[PedHash["Acult01AMO"] = 1430544400] = "Acult01AMO";
    PedHash[PedHash["Acult01AMY"] = 3043264555] = "Acult01AMY";
    PedHash[PedHash["Acult02AMO"] = 1268862154] = "Acult02AMO";
    PedHash[PedHash["Acult02AMY"] = 2162532142] = "Acult02AMY";
    PedHash[PedHash["AfriAmer01AMM"] = 3513928062] = "AfriAmer01AMM";
    PedHash[PedHash["Airhostess01SFY"] = 1567728751] = "Airhostess01SFY";
    PedHash[PedHash["AirworkerSMY"] = 1644266841] = "AirworkerSMY";
    PedHash[PedHash["Ammucity01SMY"] = 2651349821] = "Ammucity01SMY";
    PedHash[PedHash["AmmuCountrySMM"] = 233415434] = "AmmuCountrySMM";
    PedHash[PedHash["ArmBoss01GMM"] = 4058522530] = "ArmBoss01GMM";
    PedHash[PedHash["ArmGoon01GMM"] = 4255728232] = "ArmGoon01GMM";
    PedHash[PedHash["ArmGoon02GMY"] = 3310258058] = "ArmGoon02GMY";
    PedHash[PedHash["ArmLieut01GMM"] = 3882958867] = "ArmLieut01GMM";
    PedHash[PedHash["Armoured01SMM"] = 2512875213] = "Armoured01SMM";
    PedHash[PedHash["Armoured02SMM"] = 1669696074] = "Armoured02SMM";
    PedHash[PedHash["Armymech01SMY"] = 1657546978] = "Armymech01SMY";
    PedHash[PedHash["Autopsy01SMY"] = 2988916046] = "Autopsy01SMY";
    PedHash[PedHash["Autoshop01SMM"] = 68070371] = "Autoshop01SMM";
    PedHash[PedHash["Autoshop02SMM"] = 4033578141] = "Autoshop02SMM";
    PedHash[PedHash["Azteca01GMY"] = 1752208920] = "Azteca01GMY";
    PedHash[PedHash["BallaEast01GMY"] = 4096714883] = "BallaEast01GMY";
    PedHash[PedHash["BallaOrig01GMY"] = 588969535] = "BallaOrig01GMY";
    PedHash[PedHash["Ballas01GFY"] = 361513884] = "Ballas01GFY";
    PedHash[PedHash["BallaSout01GMY"] = 599294057] = "BallaSout01GMY";
    PedHash[PedHash["Barman01SMY"] = 3852538118] = "Barman01SMY";
    PedHash[PedHash["Bartender01SFY"] = 2014052797] = "Bartender01SFY";
    PedHash[PedHash["Baywatch01SFY"] = 1250841910] = "Baywatch01SFY";
    PedHash[PedHash["Baywatch01SMY"] = 189425762] = "Baywatch01SMY";
    PedHash[PedHash["Beach01AFM"] = 808859815] = "Beach01AFM";
    PedHash[PedHash["Beach01AFY"] = 3349113128] = "Beach01AFY";
    PedHash[PedHash["Beach01AMM"] = 1077785853] = "Beach01AMM";
    PedHash[PedHash["Beach01AMO"] = 2217202584] = "Beach01AMO";
    PedHash[PedHash["Beach01AMY"] = 3523131524] = "Beach01AMY";
    PedHash[PedHash["Beach02AMM"] = 2021631368] = "Beach02AMM";
    PedHash[PedHash["Beach02AMY"] = 600300561] = "Beach02AMY";
    PedHash[PedHash["Beach03AMY"] = 3886638041] = "Beach03AMY";
    PedHash[PedHash["Beachvesp01AMY"] = 2114544056] = "Beachvesp01AMY";
    PedHash[PedHash["Beachvesp02AMY"] = 3394697810] = "Beachvesp02AMY";
    PedHash[PedHash["Bevhills01AFM"] = 3188223741] = "Bevhills01AFM";
    PedHash[PedHash["Bevhills01AFY"] = 1146800212] = "Bevhills01AFY";
    PedHash[PedHash["Bevhills01AMM"] = 1423699487] = "Bevhills01AMM";
    PedHash[PedHash["Bevhills01AMY"] = 1982350912] = "Bevhills01AMY";
    PedHash[PedHash["Bevhills02AFM"] = 2688103263] = "Bevhills02AFM";
    PedHash[PedHash["Bevhills02AFY"] = 1546450936] = "Bevhills02AFY";
    PedHash[PedHash["Bevhills02AMM"] = 1068876755] = "Bevhills02AMM";
    PedHash[PedHash["Bevhills02AMY"] = 1720428295] = "Bevhills02AMY";
    PedHash[PedHash["Bevhills03AFY"] = 549978415] = "Bevhills03AFY";
    PedHash[PedHash["Bevhills04AFY"] = 920595805] = "Bevhills04AFY";
    PedHash[PedHash["Blackops01SMY"] = 3019107892] = "Blackops01SMY";
    PedHash[PedHash["Blackops02SMY"] = 2047212121] = "Blackops02SMY";
    PedHash[PedHash["Blackops03SMY"] = 1349953339] = "Blackops03SMY";
    PedHash[PedHash["Bodybuild01AFM"] = 1004114196] = "Bodybuild01AFM";
    PedHash[PedHash["Bouncer01SMM"] = 2681481517] = "Bouncer01SMM";
    PedHash[PedHash["Breakdance01AMY"] = 933205398] = "Breakdance01AMY";
    PedHash[PedHash["Busboy01SMY"] = 3640249671] = "Busboy01SMY";
    PedHash[PedHash["Busicas01AMY"] = 2597531625] = "Busicas01AMY";
    PedHash[PedHash["Business01AFY"] = 664399832] = "Business01AFY";
    PedHash[PedHash["Business01AMM"] = 2120901815] = "Business01AMM";
    PedHash[PedHash["Business01AMY"] = 3382649284] = "Business01AMY";
    PedHash[PedHash["Business02AFM"] = 532905404] = "Business02AFM";
    PedHash[PedHash["Business02AFY"] = 826475330] = "Business02AFY";
    PedHash[PedHash["Business02AMY"] = 3014915558] = "Business02AMY";
    PedHash[PedHash["Business03AFY"] = 2928082356] = "Business03AFY";
    PedHash[PedHash["Business03AMY"] = 2705543429] = "Business03AMY";
    PedHash[PedHash["Business04AFY"] = 3083210802] = "Business04AFY";
    PedHash[PedHash["Busker01SMO"] = 2912874939] = "Busker01SMO";
    PedHash[PedHash["CCrew01SMM"] = 3387290987] = "CCrew01SMM";
    PedHash[PedHash["Chef01SMY"] = 261586155] = "Chef01SMY";
    PedHash[PedHash["ChemSec01SMM"] = 788443093] = "ChemSec01SMM";
    PedHash[PedHash["ChemWork01GMM"] = 4128603535] = "ChemWork01GMM";
    PedHash[PedHash["ChiBoss01GMM"] = 3118269184] = "ChiBoss01GMM";
    PedHash[PedHash["ChiCold01GMM"] = 275618457] = "ChiCold01GMM";
    PedHash[PedHash["ChiGoon01GMM"] = 2119136831] = "ChiGoon01GMM";
    PedHash[PedHash["ChiGoon02GMM"] = 4285659174] = "ChiGoon02GMM";
    PedHash[PedHash["CiaSec01SMM"] = 1650288984] = "CiaSec01SMM";
    PedHash[PedHash["Clown01SMY"] = 71929310] = "Clown01SMY";
    PedHash[PedHash["Cntrybar01SMM"] = 436345731] = "Cntrybar01SMM";
    PedHash[PedHash["Construct01SMY"] = 3621428889] = "Construct01SMY";
    PedHash[PedHash["Construct02SMY"] = 3321821918] = "Construct02SMY";
    PedHash[PedHash["Cop01SFY"] = 368603149] = "Cop01SFY";
    PedHash[PedHash["Cop01SMY"] = 1581098148] = "Cop01SMY";
    PedHash[PedHash["Cyclist01AMY"] = 4257633223] = "Cyclist01AMY";
    PedHash[PedHash["Dealer01SMY"] = 3835149295] = "Dealer01SMY";
    PedHash[PedHash["Devinsec01SMY"] = 2606068340] = "Devinsec01SMY";
    PedHash[PedHash["Dhill01AMY"] = 4282288299] = "Dhill01AMY";
    PedHash[PedHash["Dockwork01SMM"] = 349680864] = "Dockwork01SMM";
    PedHash[PedHash["Dockwork01SMY"] = 2255894993] = "Dockwork01SMY";
    PedHash[PedHash["Doctor01SMM"] = 3564307372] = "Doctor01SMM";
    PedHash[PedHash["Doorman01SMY"] = 579932932] = "Doorman01SMY";
    PedHash[PedHash["Downtown01AFM"] = 1699403886] = "Downtown01AFM";
    PedHash[PedHash["Downtown01AMY"] = 766375082] = "Downtown01AMY";
    PedHash[PedHash["DwService01SMY"] = 1976765073] = "DwService01SMY";
    PedHash[PedHash["DwService02SMY"] = 4119890438] = "DwService02SMY";
    PedHash[PedHash["Eastsa01AFM"] = 2638072698] = "Eastsa01AFM";
    PedHash[PedHash["Eastsa01AFY"] = 4121954205] = "Eastsa01AFY";
    PedHash[PedHash["Eastsa01AMM"] = 4188468543] = "Eastsa01AMM";
    PedHash[PedHash["Eastsa01AMY"] = 2756120947] = "Eastsa01AMY";
    PedHash[PedHash["Eastsa02AFM"] = 1674107025] = "Eastsa02AFM";
    PedHash[PedHash["Eastsa02AFY"] = 70821038] = "Eastsa02AFY";
    PedHash[PedHash["Eastsa02AMM"] = 131961260] = "Eastsa02AMM";
    PedHash[PedHash["Eastsa02AMY"] = 377976310] = "Eastsa02AMY";
    PedHash[PedHash["Eastsa03AFY"] = 1371553700] = "Eastsa03AFY";
    PedHash[PedHash["Epsilon01AFY"] = 1755064960] = "Epsilon01AFY";
    PedHash[PedHash["Epsilon01AMY"] = 2010389054] = "Epsilon01AMY";
    PedHash[PedHash["Epsilon02AMY"] = 2860711835] = "Epsilon02AMY";
    PedHash[PedHash["Factory01SFY"] = 1777626099] = "Factory01SFY";
    PedHash[PedHash["Factory01SMY"] = 1097048408] = "Factory01SMY";
    PedHash[PedHash["Famca01GMY"] = 3896218551] = "Famca01GMY";
    PedHash[PedHash["Famdnf01GMY"] = 3681718840] = "Famdnf01GMY";
    PedHash[PedHash["Famfor01GMY"] = 2217749257] = "Famfor01GMY";
    PedHash[PedHash["Families01GFY"] = 1309468115] = "Families01GFY";
    PedHash[PedHash["Farmer01AMM"] = 2488675799] = "Farmer01AMM";
    PedHash[PedHash["FatBla01AFM"] = 4206136267] = "FatBla01AFM";
    PedHash[PedHash["FatCult01AFM"] = 3050275044] = "FatCult01AFM";
    PedHash[PedHash["Fatlatin01AMM"] = 1641152947] = "Fatlatin01AMM";
    PedHash[PedHash["FatWhite01AFM"] = 951767867] = "FatWhite01AFM";
    PedHash[PedHash["FemBarberSFM"] = 373000027] = "FemBarberSFM";
    PedHash[PedHash["FibOffice01SMM"] = 3988550982] = "FibOffice01SMM";
    PedHash[PedHash["FibOffice02SMM"] = 653289389] = "FibOffice02SMM";
    PedHash[PedHash["FibSec01SMM"] = 2072724299] = "FibSec01SMM";
    PedHash[PedHash["Fireman01SMY"] = 3065114024] = "Fireman01SMY";
    PedHash[PedHash["Fitness01AFY"] = 1165780219] = "Fitness01AFY";
    PedHash[PedHash["Fitness02AFY"] = 331645324] = "Fitness02AFY";
    PedHash[PedHash["Gaffer01SMM"] = 2841034142] = "Gaffer01SMM";
    PedHash[PedHash["GarbageSMY"] = 4000686095] = "GarbageSMY";
    PedHash[PedHash["Gardener01SMM"] = 1240094341] = "Gardener01SMM";
    PedHash[PedHash["Gay01AMY"] = 3519864886] = "Gay01AMY";
    PedHash[PedHash["Gay02AMY"] = 2775713665] = "Gay02AMY";
    PedHash[PedHash["Genfat01AMM"] = 115168927] = "Genfat01AMM";
    PedHash[PedHash["Genfat02AMM"] = 330231874] = "Genfat02AMM";
    PedHash[PedHash["Genhot01AFY"] = 793439294] = "Genhot01AFY";
    PedHash[PedHash["Genstreet01AFO"] = 1640504453] = "Genstreet01AFO";
    PedHash[PedHash["Genstreet01AMO"] = 2908022696] = "Genstreet01AMO";
    PedHash[PedHash["Genstreet01AMY"] = 2557996913] = "Genstreet01AMY";
    PedHash[PedHash["Genstreet02AMY"] = 891398354] = "Genstreet02AMY";
    PedHash[PedHash["GentransportSMM"] = 411102470] = "GentransportSMM";
    PedHash[PedHash["Golfer01AFY"] = 2111372120] = "Golfer01AFY";
    PedHash[PedHash["Golfer01AMM"] = 2850754114] = "Golfer01AMM";
    PedHash[PedHash["Golfer01AMY"] = 3609190705] = "Golfer01AMY";
    PedHash[PedHash["Grip01SMY"] = 815693290] = "Grip01SMY";
    PedHash[PedHash["Hairdress01SMM"] = 1099825042] = "Hairdress01SMM";
    PedHash[PedHash["Hasjew01AMM"] = 1809430156] = "Hasjew01AMM";
    PedHash[PedHash["Hasjew01AMY"] = 3782053633] = "Hasjew01AMY";
    PedHash[PedHash["Highsec01SMM"] = 4049719826] = "Highsec01SMM";
    PedHash[PedHash["Highsec02SMM"] = 691061163] = "Highsec02SMM";
    PedHash[PedHash["Hiker01AFY"] = 813893651] = "Hiker01AFY";
    PedHash[PedHash["Hiker01AMY"] = 1358380044] = "Hiker01AMY";
    PedHash[PedHash["Hillbilly01AMM"] = 1822107721] = "Hillbilly01AMM";
    PedHash[PedHash["Hillbilly02AMM"] = 2064532783] = "Hillbilly02AMM";
    PedHash[PedHash["Hippie01AFY"] = 343259175] = "Hippie01AFY";
    PedHash[PedHash["Hippy01AMY"] = 2097407511] = "Hippy01AMY";
    PedHash[PedHash["Hipster01AFY"] = 2185745201] = "Hipster01AFY";
    PedHash[PedHash["Hipster01AMY"] = 587703123] = "Hipster01AMY";
    PedHash[PedHash["Hipster02AFY"] = 2549481101] = "Hipster02AFY";
    PedHash[PedHash["Hipster02AMY"] = 349505262] = "Hipster02AMY";
    PedHash[PedHash["Hipster03AFY"] = 2780469782] = "Hipster03AFY";
    PedHash[PedHash["Hipster03AMY"] = 1312913862] = "Hipster03AMY";
    PedHash[PedHash["Hipster04AFY"] = 429425116] = "Hipster04AFY";
    PedHash[PedHash["Hooker01SFY"] = 42647445] = "Hooker01SFY";
    PedHash[PedHash["Hooker02SFY"] = 348382215] = "Hooker02SFY";
    PedHash[PedHash["Hooker03SFY"] = 51789996] = "Hooker03SFY";
    PedHash[PedHash["Hwaycop01SMY"] = 1939545845] = "Hwaycop01SMY";
    PedHash[PedHash["Indian01AFO"] = 3134700416] = "Indian01AFO";
    PedHash[PedHash["Indian01AFY"] = 153984193] = "Indian01AFY";
    PedHash[PedHash["Indian01AMM"] = 3721046572] = "Indian01AMM";
    PedHash[PedHash["Indian01AMY"] = 706935758] = "Indian01AMY";
    PedHash[PedHash["JanitorSMM"] = 2842417644] = "JanitorSMM";
    PedHash[PedHash["Jetski01AMY"] = 767028979] = "Jetski01AMY";
    PedHash[PedHash["Juggalo01AFY"] = 3675473203] = "Juggalo01AFY";
    PedHash[PedHash["Juggalo01AMY"] = 2445950508] = "Juggalo01AMY";
    PedHash[PedHash["KorBoss01GMM"] = 891945583] = "KorBoss01GMM";
    PedHash[PedHash["Korean01GMY"] = 611648169] = "Korean01GMY";
    PedHash[PedHash["Korean02GMY"] = 2414729609] = "Korean02GMY";
    PedHash[PedHash["KorLieut01GMY"] = 2093736314] = "KorLieut01GMY";
    PedHash[PedHash["Ktown01AFM"] = 1388848350] = "Ktown01AFM";
    PedHash[PedHash["Ktown01AFO"] = 1204772502] = "Ktown01AFO";
    PedHash[PedHash["Ktown01AMM"] = 3512565361] = "Ktown01AMM";
    PedHash[PedHash["Ktown01AMO"] = 355916122] = "Ktown01AMO";
    PedHash[PedHash["Ktown01AMY"] = 452351020] = "Ktown01AMY";
    PedHash[PedHash["Ktown02AFM"] = 1090617681] = "Ktown02AFM";
    PedHash[PedHash["Ktown02AMY"] = 696250687] = "Ktown02AMY";
    PedHash[PedHash["Lathandy01SMM"] = 2659242702] = "Lathandy01SMM";
    PedHash[PedHash["Latino01AMY"] = 321657486] = "Latino01AMY";
    PedHash[PedHash["Lifeinvad01SMM"] = 3724572669] = "Lifeinvad01SMM";
    PedHash[PedHash["LinecookSMM"] = 3684436375] = "LinecookSMM";
    PedHash[PedHash["Lost01GFY"] = 4250220510] = "Lost01GFY";
    PedHash[PedHash["Lost01GMY"] = 1330042375] = "Lost01GMY";
    PedHash[PedHash["Lost02GMY"] = 1032073858] = "Lost02GMY";
    PedHash[PedHash["Lost03GMY"] = 850468060] = "Lost03GMY";
    PedHash[PedHash["Lsmetro01SMM"] = 1985653476] = "Lsmetro01SMM";
    PedHash[PedHash["Maid01SFM"] = 3767780806] = "Maid01SFM";
    PedHash[PedHash["Malibu01AMM"] = 803106487] = "Malibu01AMM";
    PedHash[PedHash["Mariachi01SMM"] = 2124742566] = "Mariachi01SMM";
    PedHash[PedHash["Marine01SMM"] = 4074414829] = "Marine01SMM";
    PedHash[PedHash["Marine01SMY"] = 1702441027] = "Marine01SMY";
    PedHash[PedHash["Marine02SMM"] = 4028996995] = "Marine02SMM";
    PedHash[PedHash["Marine02SMY"] = 1490458366] = "Marine02SMY";
    PedHash[PedHash["Marine03SMY"] = 1925237458] = "Marine03SMY";
    PedHash[PedHash["Methhead01AMY"] = 1768677545] = "Methhead01AMY";
    PedHash[PedHash["MexBoss01GMM"] = 1466037421] = "MexBoss01GMM";
    PedHash[PedHash["MexBoss02GMM"] = 1226102803] = "MexBoss02GMM";
    PedHash[PedHash["MexCntry01AMM"] = 3716251309] = "MexCntry01AMM";
    PedHash[PedHash["MexGang01GMY"] = 3185399110] = "MexGang01GMY";
    PedHash[PedHash["MexGoon01GMY"] = 653210662] = "MexGoon01GMY";
    PedHash[PedHash["MexGoon02GMY"] = 832784782] = "MexGoon02GMY";
    PedHash[PedHash["MexGoon03GMY"] = 2521633500] = "MexGoon03GMY";
    PedHash[PedHash["MexLabor01AMM"] = 2992445106] = "MexLabor01AMM";
    PedHash[PedHash["MexThug01AMY"] = 810804565] = "MexThug01AMY";
    PedHash[PedHash["Migrant01SFY"] = 3579522037] = "Migrant01SFY";
    PedHash[PedHash["Migrant01SMM"] = 3977045190] = "Migrant01SMM";
    PedHash[PedHash["MimeSMY"] = 1021093698] = "MimeSMY";
    PedHash[PedHash["Motox01AMY"] = 1694362237] = "Motox01AMY";
    PedHash[PedHash["Motox02AMY"] = 2007797722] = "Motox02AMY";
    PedHash[PedHash["MovAlien01"] = 1684083350] = "MovAlien01";
    PedHash[PedHash["MovPrem01SFY"] = 587253782] = "MovPrem01SFY";
    PedHash[PedHash["Movprem01SMM"] = 3630066984] = "Movprem01SMM";
    PedHash[PedHash["Movspace01SMM"] = 3887273010] = "Movspace01SMM";
    PedHash[PedHash["Musclbeac01AMY"] = 1264920838] = "Musclbeac01AMY";
    PedHash[PedHash["Musclbeac02AMY"] = 3374523516] = "Musclbeac02AMY";
    PedHash[PedHash["OgBoss01AMM"] = 1746653202] = "OgBoss01AMM";
    PedHash[PedHash["Paparazzi01AMM"] = 3972697109] = "Paparazzi01AMM";
    PedHash[PedHash["Paramedic01SMM"] = 3008586398] = "Paramedic01SMM";
    PedHash[PedHash["PestCont01SMY"] = 1209091352] = "PestCont01SMY";
    PedHash[PedHash["Pilot01SMM"] = 3881519900] = "Pilot01SMM";
    PedHash[PedHash["Pilot01SMY"] = 2872052743] = "Pilot01SMY";
    PedHash[PedHash["Pilot02SMM"] = 4131252449] = "Pilot02SMM";
    PedHash[PedHash["PoloGoon01GMY"] = 1329576454] = "PoloGoon01GMY";
    PedHash[PedHash["PoloGoon02GMY"] = 2733138262] = "PoloGoon02GMY";
    PedHash[PedHash["Polynesian01AMM"] = 2849617566] = "Polynesian01AMM";
    PedHash[PedHash["Polynesian01AMY"] = 2206530719] = "Polynesian01AMY";
    PedHash[PedHash["Postal01SMM"] = 1650036788] = "Postal01SMM";
    PedHash[PedHash["Postal02SMM"] = 1936142927] = "Postal02SMM";
    PedHash[PedHash["Prisguard01SMM"] = 1456041926] = "Prisguard01SMM";
    PedHash[PedHash["PrisMuscl01SMY"] = 1596003233] = "PrisMuscl01SMY";
    PedHash[PedHash["Prisoner01SMY"] = 2981862233] = "Prisoner01SMY";
    PedHash[PedHash["PrologueHostage01AFM"] = 379310561] = "PrologueHostage01AFM";
    PedHash[PedHash["PrologueHostage01AMM"] = 2534589327] = "PrologueHostage01AMM";
    PedHash[PedHash["Ranger01SFY"] = 2680682039] = "Ranger01SFY";
    PedHash[PedHash["Ranger01SMY"] = 4017173934] = "Ranger01SMY";
    PedHash[PedHash["Roadcyc01AMY"] = 4116817094] = "Roadcyc01AMY";
    PedHash[PedHash["Robber01SMY"] = 3227390873] = "Robber01SMY";
    PedHash[PedHash["RsRanger01AMO"] = 1011059922] = "RsRanger01AMO";
    PedHash[PedHash["Runner01AFY"] = 3343476521] = "Runner01AFY";
    PedHash[PedHash["Runner01AMY"] = 623927022] = "Runner01AMY";
    PedHash[PedHash["Runner02AMY"] = 2218630415] = "Runner02AMY";
    PedHash[PedHash["Rurmeth01AFY"] = 1064866854] = "Rurmeth01AFY";
    PedHash[PedHash["Rurmeth01AMM"] = 1001210244] = "Rurmeth01AMM";
    PedHash[PedHash["Salton01AFM"] = 3725461865] = "Salton01AFM";
    PedHash[PedHash["Salton01AFO"] = 3439295882] = "Salton01AFO";
    PedHash[PedHash["Salton01AMM"] = 1328415626] = "Salton01AMM";
    PedHash[PedHash["Salton01AMO"] = 539004493] = "Salton01AMO";
    PedHash[PedHash["Salton01AMY"] = 3613420592] = "Salton01AMY";
    PedHash[PedHash["Salton02AMM"] = 1626646295] = "Salton02AMM";
    PedHash[PedHash["Salton03AMM"] = 2995538501] = "Salton03AMM";
    PedHash[PedHash["Salton04AMM"] = 2521108919] = "Salton04AMM";
    PedHash[PedHash["SalvaBoss01GMY"] = 2422005962] = "SalvaBoss01GMY";
    PedHash[PedHash["SalvaGoon01GMY"] = 663522487] = "SalvaGoon01GMY";
    PedHash[PedHash["SalvaGoon02GMY"] = 846439045] = "SalvaGoon02GMY";
    PedHash[PedHash["SalvaGoon03GMY"] = 62440720] = "SalvaGoon03GMY";
    PedHash[PedHash["SbikeAMO"] = 1794381917] = "SbikeAMO";
    PedHash[PedHash["Scdressy01AFY"] = 3680420864] = "Scdressy01AFY";
    PedHash[PedHash["Scientist01SMM"] = 1092080539] = "Scientist01SMM";
    PedHash[PedHash["Scrubs01SFY"] = 2874755766] = "Scrubs01SFY";
    PedHash[PedHash["Security01SMM"] = 3613962792] = "Security01SMM";
    PedHash[PedHash["Sheriff01SFY"] = 1096929346] = "Sheriff01SFY";
    PedHash[PedHash["Sheriff01SMY"] = 2974087609] = "Sheriff01SMY";
    PedHash[PedHash["ShopHighSFM"] = 2923947184] = "ShopHighSFM";
    PedHash[PedHash["ShopLowSFY"] = 2842568196] = "ShopLowSFY";
    PedHash[PedHash["ShopMaskSMY"] = 1846684678] = "ShopMaskSMY";
    PedHash[PedHash["ShopMidSFY"] = 1055701597] = "ShopMidSFY";
    PedHash[PedHash["Skater01AFY"] = 1767892582] = "Skater01AFY";
    PedHash[PedHash["Skater01AMM"] = 3654768780] = "Skater01AMM";
    PedHash[PedHash["Skater01AMY"] = 3250873975] = "Skater01AMY";
    PedHash[PedHash["Skater02AMY"] = 2952446692] = "Skater02AMY";
    PedHash[PedHash["Skidrow01AFM"] = 2962707003] = "Skidrow01AFM";
    PedHash[PedHash["Skidrow01AMM"] = 32417469] = "Skidrow01AMM";
    PedHash[PedHash["Snowcop01SMM"] = 451459928] = "Snowcop01SMM";
    PedHash[PedHash["Socenlat01AMM"] = 193817059] = "Socenlat01AMM";
    PedHash[PedHash["Soucent01AFM"] = 1951946145] = "Soucent01AFM";
    PedHash[PedHash["Soucent01AFO"] = 1039800368] = "Soucent01AFO";
    PedHash[PedHash["Soucent01AFY"] = 744758650] = "Soucent01AFY";
    PedHash[PedHash["Soucent01AMM"] = 1750583735] = "Soucent01AMM";
    PedHash[PedHash["Soucent01AMO"] = 718836251] = "Soucent01AMO";
    PedHash[PedHash["Soucent01AMY"] = 3877027275] = "Soucent01AMY";
    PedHash[PedHash["Soucent02AFM"] = 4079145784] = "Soucent02AFM";
    PedHash[PedHash["Soucent02AFO"] = 2775443222] = "Soucent02AFO";
    PedHash[PedHash["Soucent02AFY"] = 1519319503] = "Soucent02AFY";
    PedHash[PedHash["Soucent02AMM"] = 2674735073] = "Soucent02AMM";
    PedHash[PedHash["Soucent02AMO"] = 1082572151] = "Soucent02AMO";
    PedHash[PedHash["Soucent02AMY"] = 2896414922] = "Soucent02AMY";
    PedHash[PedHash["Soucent03AFY"] = 2276611093] = "Soucent03AFY";
    PedHash[PedHash["Soucent03AMM"] = 2346291386] = "Soucent03AMM";
    PedHash[PedHash["Soucent03AMO"] = 238213328] = "Soucent03AMO";
    PedHash[PedHash["Soucent03AMY"] = 3287349092] = "Soucent03AMY";
    PedHash[PedHash["Soucent04AMM"] = 3271294718] = "Soucent04AMM";
    PedHash[PedHash["Soucent04AMY"] = 2318861297] = "Soucent04AMY";
    PedHash[PedHash["Soucentmc01AFM"] = 3454621138] = "Soucentmc01AFM";
    PedHash[PedHash["Staggrm01AMO"] = 2442448387] = "Staggrm01AMO";
    PedHash[PedHash["Stbla01AMY"] = 3482496489] = "Stbla01AMY";
    PedHash[PedHash["Stbla02AMY"] = 2563194959] = "Stbla02AMY";
    PedHash[PedHash["Stlat01AMY"] = 2255803900] = "Stlat01AMY";
    PedHash[PedHash["Stlat02AMM"] = 3265820418] = "Stlat02AMM";
    PedHash[PedHash["Stripper01SFY"] = 1381498905] = "Stripper01SFY";
    PedHash[PedHash["Stripper02SFY"] = 1846523796] = "Stripper02SFY";
    PedHash[PedHash["StripperLiteSFY"] = 1544875514] = "StripperLiteSFY";
    PedHash[PedHash["Strperf01SMM"] = 2035992488] = "Strperf01SMM";
    PedHash[PedHash["Strpreach01SMM"] = 469792763] = "Strpreach01SMM";
    PedHash[PedHash["StrPunk01GMY"] = 4246489531] = "StrPunk01GMY";
    PedHash[PedHash["StrPunk02GMY"] = 228715206] = "StrPunk02GMY";
    PedHash[PedHash["Strvend01SMM"] = 3465614249] = "Strvend01SMM";
    PedHash[PedHash["Strvend01SMY"] = 2457805603] = "Strvend01SMY";
    PedHash[PedHash["Stwhi01AMY"] = 605602864] = "Stwhi01AMY";
    PedHash[PedHash["Stwhi02AMY"] = 919005580] = "Stwhi02AMY";
    PedHash[PedHash["Sunbathe01AMY"] = 3072929548] = "Sunbathe01AMY";
    PedHash[PedHash["Surfer01AMY"] = 3938633710] = "Surfer01AMY";
    PedHash[PedHash["Swat01SMY"] = 2374966032] = "Swat01SMY";
    PedHash[PedHash["Sweatshop01SFM"] = 824925120] = "Sweatshop01SFM";
    PedHash[PedHash["Sweatshop01SFY"] = 2231547570] = "Sweatshop01SFY";
    PedHash[PedHash["Tattoo01AMO"] = 2494442380] = "Tattoo01AMO";
    PedHash[PedHash["Tennis01AFY"] = 1426880966] = "Tennis01AFY";
    PedHash[PedHash["Tennis01AMM"] = 1416254276] = "Tennis01AMM";
    PedHash[PedHash["Topless01AFY"] = 2633130371] = "Topless01AFY";
    PedHash[PedHash["Tourist01AFM"] = 1347814329] = "Tourist01AFM";
    PedHash[PedHash["Tourist01AFY"] = 1446741360] = "Tourist01AFY";
    PedHash[PedHash["Tourist01AMM"] = 3365863812] = "Tourist01AMM";
    PedHash[PedHash["Tourist02AFY"] = 2435054400] = "Tourist02AFY";
    PedHash[PedHash["Tramp01AFM"] = 1224306523] = "Tramp01AFM";
    PedHash[PedHash["Tramp01AMM"] = 516505552] = "Tramp01AMM";
    PedHash[PedHash["Tramp01AMO"] = 390939205] = "Tramp01AMO";
    PedHash[PedHash["TrampBeac01AFM"] = 2359345766] = "TrampBeac01AFM";
    PedHash[PedHash["TrampBeac01AMM"] = 1404403376] = "TrampBeac01AMM";
    PedHash[PedHash["Tranvest01AMM"] = 3773208948] = "Tranvest01AMM";
    PedHash[PedHash["Tranvest02AMM"] = 4144940484] = "Tranvest02AMM";
    PedHash[PedHash["Trucker01SMM"] = 1498487404] = "Trucker01SMM";
    PedHash[PedHash["Ups01SMM"] = 2680389410] = "Ups01SMM";
    PedHash[PedHash["Ups02SMM"] = 3502104854] = "Ups02SMM";
    PedHash[PedHash["Uscg01SMY"] = 3389018345] = "Uscg01SMY";
    PedHash[PedHash["Vagos01GFY"] = 1520708641] = "Vagos01GFY";
    PedHash[PedHash["Valet01SMY"] = 999748158] = "Valet01SMY";
    PedHash[PedHash["Vindouche01AMY"] = 3247667175] = "Vindouche01AMY";
    PedHash[PedHash["Vinewood01AFY"] = 435429221] = "Vinewood01AFY";
    PedHash[PedHash["Vinewood01AMY"] = 1264851357] = "Vinewood01AMY";
    PedHash[PedHash["Vinewood02AFY"] = 3669401835] = "Vinewood02AFY";
    PedHash[PedHash["Vinewood02AMY"] = 1561705728] = "Vinewood02AMY";
    PedHash[PedHash["Vinewood03AFY"] = 933092024] = "Vinewood03AFY";
    PedHash[PedHash["Vinewood03AMY"] = 534725268] = "Vinewood03AMY";
    PedHash[PedHash["Vinewood04AFY"] = 4209271110] = "Vinewood04AFY";
    PedHash[PedHash["Vinewood04AMY"] = 835315305] = "Vinewood04AMY";
    PedHash[PedHash["Waiter01SMY"] = 2907468364] = "Waiter01SMY";
    PedHash[PedHash["WinClean01SMY"] = 1426951581] = "WinClean01SMY";
    PedHash[PedHash["Xmech01SMY"] = 1142162924] = "Xmech01SMY";
    PedHash[PedHash["Xmech02SMY"] = 3189832196] = "Xmech02SMY";
    PedHash[PedHash["Xmech02SMYMP"] = 1755203590] = "Xmech02SMYMP";
    PedHash[PedHash["Yoga01AFY"] = 3290105390] = "Yoga01AFY";
    PedHash[PedHash["Yoga01AMY"] = 2869588309] = "Yoga01AMY";
})(PedHash = exports.PedHash || (exports.PedHash = {}));
