// fivem-js/lib/hashes/VehicleHash.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VehicleHash = void 0;
var VehicleHash;
(function (VehicleHash) {
    VehicleHash[VehicleHash["Adder"] = 3078201489] = "Adder";
    VehicleHash[VehicleHash["Airbus"] = 1283517198] = "Airbus";
    VehicleHash[VehicleHash["Airtug"] = 1560980623] = "Airtug";
    VehicleHash[VehicleHash["Akuma"] = 1672195559] = "Akuma";
    VehicleHash[VehicleHash["Alpha"] = 767087018] = "Alpha";
    VehicleHash[VehicleHash["Ambulance"] = 1171614426] = "Ambulance";
    VehicleHash[VehicleHash["Annihilator"] = 837858166] = "Annihilator";
    VehicleHash[VehicleHash["Apc"] = 562680400] = "Apc";
    VehicleHash[VehicleHash["Ardent"] = 159274291] = "Ardent";
    VehicleHash[VehicleHash["ArmyTanker"] = 3087536137] = "ArmyTanker";
    VehicleHash[VehicleHash["ArmyTrailer"] = 2818520053] = "ArmyTrailer";
    VehicleHash[VehicleHash["ArmyTrailer2"] = 2657817814] = "ArmyTrailer2";
    VehicleHash[VehicleHash["Asea"] = 2485144969] = "Asea";
    VehicleHash[VehicleHash["Asea2"] = 2487343317] = "Asea2";
    VehicleHash[VehicleHash["Asterope"] = 2391954683] = "Asterope";
    VehicleHash[VehicleHash["Avarus"] = 2179174271] = "Avarus";
    VehicleHash[VehicleHash["Bagger"] = 2154536131] = "Bagger";
    VehicleHash[VehicleHash["BaleTrailer"] = 3895125590] = "BaleTrailer";
    VehicleHash[VehicleHash["Baller"] = 3486135912] = "Baller";
    VehicleHash[VehicleHash["Baller2"] = 142944341] = "Baller2";
    VehicleHash[VehicleHash["Baller3"] = 1878062887] = "Baller3";
    VehicleHash[VehicleHash["Baller4"] = 634118882] = "Baller4";
    VehicleHash[VehicleHash["Baller5"] = 470404958] = "Baller5";
    VehicleHash[VehicleHash["Baller6"] = 666166960] = "Baller6";
    VehicleHash[VehicleHash["Banshee"] = 3253274834] = "Banshee";
    VehicleHash[VehicleHash["Banshee2"] = 633712403] = "Banshee2";
    VehicleHash[VehicleHash["Barracks"] = 3471458123] = "Barracks";
    VehicleHash[VehicleHash["Barracks2"] = 1074326203] = "Barracks2";
    VehicleHash[VehicleHash["Barracks3"] = 630371791] = "Barracks3";
    VehicleHash[VehicleHash["Bati"] = 4180675781] = "Bati";
    VehicleHash[VehicleHash["Bati2"] = 3403504941] = "Bati2";
    VehicleHash[VehicleHash["Benson"] = 2053223216] = "Benson";
    VehicleHash[VehicleHash["Besra"] = 1824333165] = "Besra";
    VehicleHash[VehicleHash["BestiaGTS"] = 1274868363] = "BestiaGTS";
    VehicleHash[VehicleHash["BF400"] = 86520421] = "BF400";
    VehicleHash[VehicleHash["BfInjection"] = 1126868326] = "BfInjection";
    VehicleHash[VehicleHash["Biff"] = 850991848] = "Biff";
    VehicleHash[VehicleHash["Bifta"] = 3945366167] = "Bifta";
    VehicleHash[VehicleHash["Bison"] = 4278019151] = "Bison";
    VehicleHash[VehicleHash["Bison2"] = 2072156101] = "Bison2";
    VehicleHash[VehicleHash["Bison3"] = 1739845664] = "Bison3";
    VehicleHash[VehicleHash["BJXL"] = 850565707] = "BJXL";
    VehicleHash[VehicleHash["Blade"] = 3089165662] = "Blade";
    VehicleHash[VehicleHash["Blazer"] = 2166734073] = "Blazer";
    VehicleHash[VehicleHash["Blazer2"] = 4246935337] = "Blazer2";
    VehicleHash[VehicleHash["Blazer3"] = 3025077634] = "Blazer3";
    VehicleHash[VehicleHash["Blazer4"] = 3854198872] = "Blazer4";
    VehicleHash[VehicleHash["Blazer5"] = 2704629607] = "Blazer5";
    VehicleHash[VehicleHash["Blimp"] = 4143991942] = "Blimp";
    VehicleHash[VehicleHash["Blimp2"] = 3681241380] = "Blimp2";
    VehicleHash[VehicleHash["Blista"] = 3950024287] = "Blista";
    VehicleHash[VehicleHash["Blista2"] = 1039032026] = "Blista2";
    VehicleHash[VehicleHash["Blista3"] = 3703315515] = "Blista3";
    VehicleHash[VehicleHash["Bmx"] = 1131912276] = "Bmx";
    VehicleHash[VehicleHash["BoatTrailer"] = 524108981] = "BoatTrailer";
    VehicleHash[VehicleHash["BobcatXL"] = 1069929536] = "BobcatXL";
    VehicleHash[VehicleHash["Bodhi2"] = 2859047862] = "Bodhi2";
    VehicleHash[VehicleHash["Boxville"] = 2307837162] = "Boxville";
    VehicleHash[VehicleHash["Boxville2"] = 4061868990] = "Boxville2";
    VehicleHash[VehicleHash["Boxville3"] = 121658888] = "Boxville3";
    VehicleHash[VehicleHash["Boxville4"] = 444171386] = "Boxville4";
    VehicleHash[VehicleHash["Boxville5"] = 682434785] = "Boxville5";
    VehicleHash[VehicleHash["Brawler"] = 2815302597] = "Brawler";
    VehicleHash[VehicleHash["Brickade"] = 3989239879] = "Brickade";
    VehicleHash[VehicleHash["Brioso"] = 1549126457] = "Brioso";
    VehicleHash[VehicleHash["BType"] = 117401876] = "BType";
    VehicleHash[VehicleHash["BType2"] = 3463132580] = "BType2";
    VehicleHash[VehicleHash["BType3"] = 3692679425] = "BType3";
    VehicleHash[VehicleHash["Buccaneer"] = 3612755468] = "Buccaneer";
    VehicleHash[VehicleHash["Buccaneer2"] = 3281516360] = "Buccaneer2";
    VehicleHash[VehicleHash["Buffalo"] = 3990165190] = "Buffalo";
    VehicleHash[VehicleHash["Buffalo2"] = 736902334] = "Buffalo2";
    VehicleHash[VehicleHash["Buffalo3"] = 237764926] = "Buffalo3";
    VehicleHash[VehicleHash["Bulldozer"] = 1886712733] = "Bulldozer";
    VehicleHash[VehicleHash["Bullet"] = 2598821281] = "Bullet";
    VehicleHash[VehicleHash["Burrito"] = 2948279460] = "Burrito";
    VehicleHash[VehicleHash["Burrito2"] = 3387490166] = "Burrito2";
    VehicleHash[VehicleHash["Burrito3"] = 2551651283] = "Burrito3";
    VehicleHash[VehicleHash["Burrito4"] = 893081117] = "Burrito4";
    VehicleHash[VehicleHash["Burrito5"] = 1132262048] = "Burrito5";
    VehicleHash[VehicleHash["Bus"] = 3581397346] = "Bus";
    VehicleHash[VehicleHash["Buzzard"] = 788747387] = "Buzzard";
    VehicleHash[VehicleHash["Buzzard2"] = 745926877] = "Buzzard2";
    VehicleHash[VehicleHash["CableCar"] = 3334677549] = "CableCar";
    VehicleHash[VehicleHash["Caddy"] = 1147287684] = "Caddy";
    VehicleHash[VehicleHash["Caddy2"] = 3757070668] = "Caddy2";
    VehicleHash[VehicleHash["Caddy3"] = 3525819835] = "Caddy3";
    VehicleHash[VehicleHash["Camper"] = 1876516712] = "Camper";
    VehicleHash[VehicleHash["Carbonizzare"] = 2072687711] = "Carbonizzare";
    VehicleHash[VehicleHash["CarbonRS"] = 11251904] = "CarbonRS";
    VehicleHash[VehicleHash["Cargobob"] = 4244420235] = "Cargobob";
    VehicleHash[VehicleHash["Cargobob2"] = 1621617168] = "Cargobob2";
    VehicleHash[VehicleHash["Cargobob3"] = 1394036463] = "Cargobob3";
    VehicleHash[VehicleHash["Cargobob4"] = 2025593404] = "Cargobob4";
    VehicleHash[VehicleHash["CargoPlane"] = 368211810] = "CargoPlane";
    VehicleHash[VehicleHash["Casco"] = 941800958] = "Casco";
    VehicleHash[VehicleHash["Cavalcade"] = 2006918058] = "Cavalcade";
    VehicleHash[VehicleHash["Cavalcade2"] = 3505073125] = "Cavalcade2";
    VehicleHash[VehicleHash["Cheetah"] = 2983812512] = "Cheetah";
    VehicleHash[VehicleHash["Cheetah2"] = 223240013] = "Cheetah2";
    VehicleHash[VehicleHash["Chimera"] = 6774487] = "Chimera";
    VehicleHash[VehicleHash["Chino"] = 349605904] = "Chino";
    VehicleHash[VehicleHash["Chino2"] = 2933279331] = "Chino2";
    VehicleHash[VehicleHash["Cliffhanger"] = 390201602] = "Cliffhanger";
    VehicleHash[VehicleHash["Coach"] = 2222034228] = "Coach";
    VehicleHash[VehicleHash["Cog55"] = 906642318] = "Cog55";
    VehicleHash[VehicleHash["Cog552"] = 704435172] = "Cog552";
    VehicleHash[VehicleHash["CogCabrio"] = 330661258] = "CogCabrio";
    VehicleHash[VehicleHash["Cognoscenti"] = 2264796000] = "Cognoscenti";
    VehicleHash[VehicleHash["Cognoscenti2"] = 3690124666] = "Cognoscenti2";
    VehicleHash[VehicleHash["Comet2"] = 3249425686] = "Comet2";
    VehicleHash[VehicleHash["Comet3"] = 2272483501] = "Comet3";
    VehicleHash[VehicleHash["Contender"] = 683047626] = "Contender";
    VehicleHash[VehicleHash["Coquette"] = 108773431] = "Coquette";
    VehicleHash[VehicleHash["Coquette2"] = 1011753235] = "Coquette2";
    VehicleHash[VehicleHash["Coquette3"] = 784565758] = "Coquette3";
    VehicleHash[VehicleHash["Cruiser"] = 448402357] = "Cruiser";
    VehicleHash[VehicleHash["Crusader"] = 321739290] = "Crusader";
    VehicleHash[VehicleHash["Cuban800"] = 3650256867] = "Cuban800";
    VehicleHash[VehicleHash["Cutter"] = 3288047904] = "Cutter";
    VehicleHash[VehicleHash["Daemon"] = 2006142190] = "Daemon";
    VehicleHash[VehicleHash["Daemon2"] = 2890830793] = "Daemon2";
    VehicleHash[VehicleHash["Defiler"] = 822018448] = "Defiler";
    VehicleHash[VehicleHash["Diablous"] = 4055125828] = "Diablous";
    VehicleHash[VehicleHash["Diablous2"] = 1790834270] = "Diablous2";
    VehicleHash[VehicleHash["Dilettante"] = 3164157193] = "Dilettante";
    VehicleHash[VehicleHash["Dilettante2"] = 1682114128] = "Dilettante2";
    VehicleHash[VehicleHash["Dinghy"] = 1033245328] = "Dinghy";
    VehicleHash[VehicleHash["Dinghy2"] = 276773164] = "Dinghy2";
    VehicleHash[VehicleHash["Dinghy3"] = 509498602] = "Dinghy3";
    VehicleHash[VehicleHash["Dinghy4"] = 867467158] = "Dinghy4";
    VehicleHash[VehicleHash["DLoader"] = 1770332643] = "DLoader";
    VehicleHash[VehicleHash["DockTrailer"] = 2154757102] = "DockTrailer";
    VehicleHash[VehicleHash["Docktug"] = 3410276810] = "Docktug";
    VehicleHash[VehicleHash["Dodo"] = 3393804037] = "Dodo";
    VehicleHash[VehicleHash["Dominator"] = 80636076] = "Dominator";
    VehicleHash[VehicleHash["Dominator2"] = 3379262425] = "Dominator2";
    VehicleHash[VehicleHash["Double"] = 2623969160] = "Double";
    VehicleHash[VehicleHash["Dubsta"] = 1177543287] = "Dubsta";
    VehicleHash[VehicleHash["Dubsta2"] = 3900892662] = "Dubsta2";
    VehicleHash[VehicleHash["Dubsta3"] = 3057713523] = "Dubsta3";
    VehicleHash[VehicleHash["Dukes"] = 723973206] = "Dukes";
    VehicleHash[VehicleHash["Dukes2"] = 3968823444] = "Dukes2";
    VehicleHash[VehicleHash["Dump"] = 2164484578] = "Dump";
    VehicleHash[VehicleHash["Dune"] = 2633113103] = "Dune";
    VehicleHash[VehicleHash["Dune2"] = 534258863] = "Dune2";
    VehicleHash[VehicleHash["Dune3"] = 1897744184] = "Dune3";
    VehicleHash[VehicleHash["Dune4"] = 3467805257] = "Dune4";
    VehicleHash[VehicleHash["Dune5"] = 3982671785] = "Dune5";
    VehicleHash[VehicleHash["Duster"] = 970356638] = "Duster";
    VehicleHash[VehicleHash["Elegy"] = 196747873] = "Elegy";
    VehicleHash[VehicleHash["Elegy2"] = 3728579874] = "Elegy2";
    VehicleHash[VehicleHash["Emperor"] = 3609690755] = "Emperor";
    VehicleHash[VehicleHash["Emperor2"] = 2411965148] = "Emperor2";
    VehicleHash[VehicleHash["Emperor3"] = 3053254478] = "Emperor3";
    VehicleHash[VehicleHash["Enduro"] = 1753414259] = "Enduro";
    VehicleHash[VehicleHash["EntityXF"] = 3003014393] = "EntityXF";
    VehicleHash[VehicleHash["Esskey"] = 2035069708] = "Esskey";
    VehicleHash[VehicleHash["Exemplar"] = 4289813342] = "Exemplar";
    VehicleHash[VehicleHash["F620"] = 3703357000] = "F620";
    VehicleHash[VehicleHash["Faction"] = 2175389151] = "Faction";
    VehicleHash[VehicleHash["Faction2"] = 2504420315] = "Faction2";
    VehicleHash[VehicleHash["Faction3"] = 2255212070] = "Faction3";
    VehicleHash[VehicleHash["Faggio"] = 2452219115] = "Faggio";
    VehicleHash[VehicleHash["Faggio2"] = 55628203] = "Faggio2";
    VehicleHash[VehicleHash["Faggio3"] = 3005788552] = "Faggio3";
    VehicleHash[VehicleHash["FBI"] = 1127131465] = "FBI";
    VehicleHash[VehicleHash["FBI2"] = 2647026068] = "FBI2";
    VehicleHash[VehicleHash["FCR"] = 627535535] = "FCR";
    VehicleHash[VehicleHash["FCR2"] = 3537231886] = "FCR2";
    VehicleHash[VehicleHash["Felon"] = 3903372712] = "Felon";
    VehicleHash[VehicleHash["Felon2"] = 4205676014] = "Felon2";
    VehicleHash[VehicleHash["Feltzer2"] = 2299640309] = "Feltzer2";
    VehicleHash[VehicleHash["Feltzer3"] = 2728226064] = "Feltzer3";
    VehicleHash[VehicleHash["FireTruk"] = 1938952078] = "FireTruk";
    VehicleHash[VehicleHash["Fixter"] = 3458454463] = "Fixter";
    VehicleHash[VehicleHash["Flatbed"] = 1353720154] = "Flatbed";
    VehicleHash[VehicleHash["Forklift"] = 1491375716] = "Forklift";
    VehicleHash[VehicleHash["FMJ"] = 1426219628] = "FMJ";
    VehicleHash[VehicleHash["FQ2"] = 3157435195] = "FQ2";
    VehicleHash[VehicleHash["Freight"] = 1030400667] = "Freight";
    VehicleHash[VehicleHash["FreightCar"] = 184361638] = "FreightCar";
    VehicleHash[VehicleHash["FreightCont1"] = 920453016] = "FreightCont1";
    VehicleHash[VehicleHash["FreightCont2"] = 240201337] = "FreightCont2";
    VehicleHash[VehicleHash["FreightGrain"] = 642617954] = "FreightGrain";
    VehicleHash[VehicleHash["FreightTrailer"] = 3517691494] = "FreightTrailer";
    VehicleHash[VehicleHash["Frogger"] = 744705981] = "Frogger";
    VehicleHash[VehicleHash["Frogger2"] = 1949211328] = "Frogger2";
    VehicleHash[VehicleHash["Fugitive"] = 1909141499] = "Fugitive";
    VehicleHash[VehicleHash["Furoregt"] = 3205927392] = "Furoregt";
    VehicleHash[VehicleHash["Fusilade"] = 499169875] = "Fusilade";
    VehicleHash[VehicleHash["Futo"] = 2016857647] = "Futo";
    VehicleHash[VehicleHash["Gargoyle"] = 741090084] = "Gargoyle";
    VehicleHash[VehicleHash["Gauntlet"] = 2494797253] = "Gauntlet";
    VehicleHash[VehicleHash["Gauntlet2"] = 349315417] = "Gauntlet2";
    VehicleHash[VehicleHash["GBurrito"] = 2549763894] = "GBurrito";
    VehicleHash[VehicleHash["GBurrito2"] = 296357396] = "GBurrito2";
    VehicleHash[VehicleHash["Glendale"] = 75131841] = "Glendale";
    VehicleHash[VehicleHash["GP1"] = 1234311532] = "GP1";
    VehicleHash[VehicleHash["GrainTrailer"] = 1019737494] = "GrainTrailer";
    VehicleHash[VehicleHash["Granger"] = 2519238556] = "Granger";
    VehicleHash[VehicleHash["Gresley"] = 2751205197] = "Gresley";
    VehicleHash[VehicleHash["Guardian"] = 2186977100] = "Guardian";
    VehicleHash[VehicleHash["Habanero"] = 884422927] = "Habanero";
    VehicleHash[VehicleHash["Hakuchou"] = 1265391242] = "Hakuchou";
    VehicleHash[VehicleHash["Hakuchou2"] = 4039289119] = "Hakuchou2";
    VehicleHash[VehicleHash["HalfTrack"] = 4262731174] = "HalfTrack";
    VehicleHash[VehicleHash["Handler"] = 444583674] = "Handler";
    VehicleHash[VehicleHash["Hauler"] = 1518533038] = "Hauler";
    VehicleHash[VehicleHash["Hauler2"] = 387748548] = "Hauler2";
    VehicleHash[VehicleHash["Hexer"] = 301427732] = "Hexer";
    VehicleHash[VehicleHash["Hotknife"] = 37348240] = "Hotknife";
    VehicleHash[VehicleHash["Huntley"] = 486987393] = "Huntley";
    VehicleHash[VehicleHash["Hydra"] = 970385471] = "Hydra";
    VehicleHash[VehicleHash["Infernus"] = 418536135] = "Infernus";
    VehicleHash[VehicleHash["Infernus2"] = 2889029532] = "Infernus2";
    VehicleHash[VehicleHash["Ingot"] = 3005245074] = "Ingot";
    VehicleHash[VehicleHash["Innovation"] = 4135840458] = "Innovation";
    VehicleHash[VehicleHash["Insurgent"] = 2434067162] = "Insurgent";
    VehicleHash[VehicleHash["Insurgent2"] = 2071877360] = "Insurgent2";
    VehicleHash[VehicleHash["Insurgent3"] = 2370534026] = "Insurgent3";
    VehicleHash[VehicleHash["Intruder"] = 886934177] = "Intruder";
    VehicleHash[VehicleHash["Issi2"] = 3117103977] = "Issi2";
    VehicleHash[VehicleHash["ItaliGTB"] = 2246633323] = "ItaliGTB";
    VehicleHash[VehicleHash["ItaliGTB2"] = 3812247419] = "ItaliGTB2";
    VehicleHash[VehicleHash["Jackal"] = 3670438162] = "Jackal";
    VehicleHash[VehicleHash["JB700"] = 1051415893] = "JB700";
    VehicleHash[VehicleHash["Jester"] = 2997294755] = "Jester";
    VehicleHash[VehicleHash["Jester2"] = 3188613414] = "Jester2";
    VehicleHash[VehicleHash["Jet"] = 1058115860] = "Jet";
    VehicleHash[VehicleHash["Jetmax"] = 861409633] = "Jetmax";
    VehicleHash[VehicleHash["Journey"] = 4174679674] = "Journey";
    VehicleHash[VehicleHash["Kalahari"] = 92612664] = "Kalahari";
    VehicleHash[VehicleHash["Khamelion"] = 544021352] = "Khamelion";
    VehicleHash[VehicleHash["Kuruma"] = 2922118804] = "Kuruma";
    VehicleHash[VehicleHash["Kuruma2"] = 410882957] = "Kuruma2";
    VehicleHash[VehicleHash["Landstalker"] = 1269098716] = "Landstalker";
    VehicleHash[VehicleHash["Lazer"] = 3013282534] = "Lazer";
    VehicleHash[VehicleHash["LE7B"] = 3062131285] = "LE7B";
    VehicleHash[VehicleHash["Lectro"] = 640818791] = "Lectro";
    VehicleHash[VehicleHash["Lguard"] = 469291905] = "Lguard";
    VehicleHash[VehicleHash["Limo2"] = 4180339789] = "Limo2";
    VehicleHash[VehicleHash["Lurcher"] = 2068293287] = "Lurcher";
    VehicleHash[VehicleHash["Luxor"] = 621481054] = "Luxor";
    VehicleHash[VehicleHash["Luxor2"] = 3080673438] = "Luxor2";
    VehicleHash[VehicleHash["Lynx"] = 482197771] = "Lynx";
    VehicleHash[VehicleHash["Mamba"] = 2634021974] = "Mamba";
    VehicleHash[VehicleHash["Mammatus"] = 2548391185] = "Mammatus";
    VehicleHash[VehicleHash["Manana"] = 2170765704] = "Manana";
    VehicleHash[VehicleHash["Manchez"] = 2771538552] = "Manchez";
    VehicleHash[VehicleHash["Marquis"] = 3251507587] = "Marquis";
    VehicleHash[VehicleHash["Marshall"] = 1233534620] = "Marshall";
    VehicleHash[VehicleHash["Massacro"] = 4152024626] = "Massacro";
    VehicleHash[VehicleHash["Massacro2"] = 3663206819] = "Massacro2";
    VehicleHash[VehicleHash["Maverick"] = 2634305738] = "Maverick";
    VehicleHash[VehicleHash["Mesa"] = 914654722] = "Mesa";
    VehicleHash[VehicleHash["Mesa2"] = 3546958660] = "Mesa2";
    VehicleHash[VehicleHash["Mesa3"] = 2230595153] = "Mesa3";
    VehicleHash[VehicleHash["MetroTrain"] = 868868440] = "MetroTrain";
    VehicleHash[VehicleHash["Miljet"] = 165154707] = "Miljet";
    VehicleHash[VehicleHash["Minivan"] = 3984502180] = "Minivan";
    VehicleHash[VehicleHash["Minivan2"] = 3168702960] = "Minivan2";
    VehicleHash[VehicleHash["Mixer"] = 3510150843] = "Mixer";
    VehicleHash[VehicleHash["Mixer2"] = 475220373] = "Mixer2";
    VehicleHash[VehicleHash["Monroe"] = 3861591579] = "Monroe";
    VehicleHash[VehicleHash["Monster"] = 3449006043] = "Monster";
    VehicleHash[VehicleHash["Moonbeam"] = 525509695] = "Moonbeam";
    VehicleHash[VehicleHash["Moonbeam2"] = 1896491931] = "Moonbeam2";
    VehicleHash[VehicleHash["Mower"] = 1783355638] = "Mower";
    VehicleHash[VehicleHash["Mule"] = 904750859] = "Mule";
    VehicleHash[VehicleHash["Mule2"] = 3244501995] = "Mule2";
    VehicleHash[VehicleHash["Mule3"] = 2242229361] = "Mule3";
    VehicleHash[VehicleHash["Nemesis"] = 3660088182] = "Nemesis";
    VehicleHash[VehicleHash["Nero"] = 1034187331] = "Nero";
    VehicleHash[VehicleHash["Nero2"] = 1093792632] = "Nero2";
    VehicleHash[VehicleHash["Nightblade"] = 2688780135] = "Nightblade";
    VehicleHash[VehicleHash["Nightshade"] = 2351681756] = "Nightshade";
    VehicleHash[VehicleHash["NightShark"] = 433954513] = "NightShark";
    VehicleHash[VehicleHash["Nimbus"] = 2999939664] = "Nimbus";
    VehicleHash[VehicleHash["Ninef"] = 1032823388] = "Ninef";
    VehicleHash[VehicleHash["Ninef2"] = 2833484545] = "Ninef2";
    VehicleHash[VehicleHash["Omnis"] = 3517794615] = "Omnis";
    VehicleHash[VehicleHash["Oppressor"] = 884483972] = "Oppressor";
    VehicleHash[VehicleHash["Oracle"] = 1348744438] = "Oracle";
    VehicleHash[VehicleHash["Oracle2"] = 3783366066] = "Oracle2";
    VehicleHash[VehicleHash["Osiris"] = 1987142870] = "Osiris";
    VehicleHash[VehicleHash["Packer"] = 569305213] = "Packer";
    VehicleHash[VehicleHash["Panto"] = 3863274624] = "Panto";
    VehicleHash[VehicleHash["Paradise"] = 1488164764] = "Paradise";
    VehicleHash[VehicleHash["Patriot"] = 3486509883] = "Patriot";
    VehicleHash[VehicleHash["PBus"] = 2287941233] = "PBus";
    VehicleHash[VehicleHash["PCJ"] = 3385765638] = "PCJ";
    VehicleHash[VehicleHash["Penetrator"] = 2536829930] = "Penetrator";
    VehicleHash[VehicleHash["Penumbra"] = 3917501776] = "Penumbra";
    VehicleHash[VehicleHash["Peyote"] = 1830407356] = "Peyote";
    VehicleHash[VehicleHash["Pfister811"] = 2465164804] = "Pfister811";
    VehicleHash[VehicleHash["Phantom"] = 2157618379] = "Phantom";
    VehicleHash[VehicleHash["Phantom2"] = 2645431192] = "Phantom2";
    VehicleHash[VehicleHash["Phantom3"] = 177270108] = "Phantom3";
    VehicleHash[VehicleHash["Phoenix"] = 2199527893] = "Phoenix";
    VehicleHash[VehicleHash["Picador"] = 1507916787] = "Picador";
    VehicleHash[VehicleHash["Pigalle"] = 1078682497] = "Pigalle";
    VehicleHash[VehicleHash["Police"] = 2046537925] = "Police";
    VehicleHash[VehicleHash["Police2"] = 2667966721] = "Police2";
    VehicleHash[VehicleHash["Police3"] = 1912215274] = "Police3";
    VehicleHash[VehicleHash["Police4"] = 2321795001] = "Police4";
    VehicleHash[VehicleHash["Policeb"] = 4260343491] = "Policeb";
    VehicleHash[VehicleHash["PoliceOld1"] = 2758042359] = "PoliceOld1";
    VehicleHash[VehicleHash["PoliceOld2"] = 2515846680] = "PoliceOld2";
    VehicleHash[VehicleHash["PoliceT"] = 456714581] = "PoliceT";
    VehicleHash[VehicleHash["Polmav"] = 353883353] = "Polmav";
    VehicleHash[VehicleHash["Pony"] = 4175309224] = "Pony";
    VehicleHash[VehicleHash["Pony2"] = 943752001] = "Pony2";
    VehicleHash[VehicleHash["Pounder"] = 2112052861] = "Pounder";
    VehicleHash[VehicleHash["Prairie"] = 2844316578] = "Prairie";
    VehicleHash[VehicleHash["Pranger"] = 741586030] = "Pranger";
    VehicleHash[VehicleHash["Predator"] = 3806844075] = "Predator";
    VehicleHash[VehicleHash["Premier"] = 2411098011] = "Premier";
    VehicleHash[VehicleHash["Primo"] = 3144368207] = "Primo";
    VehicleHash[VehicleHash["Primo2"] = 2254540506] = "Primo2";
    VehicleHash[VehicleHash["PropTrailer"] = 356391690] = "PropTrailer";
    VehicleHash[VehicleHash["Prototipo"] = 2123327359] = "Prototipo";
    VehicleHash[VehicleHash["Radi"] = 2643899483] = "Radi";
    VehicleHash[VehicleHash["RakeTrailer"] = 390902130] = "RakeTrailer";
    VehicleHash[VehicleHash["RancherXL"] = 1645267888] = "RancherXL";
    VehicleHash[VehicleHash["RancherXL2"] = 1933662059] = "RancherXL2";
    VehicleHash[VehicleHash["RallyTruck"] = 2191146052] = "RallyTruck";
    VehicleHash[VehicleHash["RapidGT"] = 2360515092] = "RapidGT";
    VehicleHash[VehicleHash["RapidGT2"] = 1737773231] = "RapidGT2";
    VehicleHash[VehicleHash["Raptor"] = 3620039993] = "Raptor";
    VehicleHash[VehicleHash["RatBike"] = 1873600305] = "RatBike";
    VehicleHash[VehicleHash["RatLoader"] = 3627815886] = "RatLoader";
    VehicleHash[VehicleHash["RatLoader2"] = 3705788919] = "RatLoader2";
    VehicleHash[VehicleHash["Reaper"] = 234062309] = "Reaper";
    VehicleHash[VehicleHash["Rebel"] = 3087195462] = "Rebel";
    VehicleHash[VehicleHash["Rebel2"] = 2249373259] = "Rebel2";
    VehicleHash[VehicleHash["Regina"] = 4280472072] = "Regina";
    VehicleHash[VehicleHash["RentalBus"] = 3196165219] = "RentalBus";
    VehicleHash[VehicleHash["Rhapsody"] = 841808271] = "Rhapsody";
    VehicleHash[VehicleHash["Rhino"] = 782665360] = "Rhino";
    VehicleHash[VehicleHash["Riot"] = 3089277354] = "Riot";
    VehicleHash[VehicleHash["Ripley"] = 3448987385] = "Ripley";
    VehicleHash[VehicleHash["Rocoto"] = 2136773105] = "Rocoto";
    VehicleHash[VehicleHash["Romero"] = 627094268] = "Romero";
    VehicleHash[VehicleHash["Rubble"] = 2589662668] = "Rubble";
    VehicleHash[VehicleHash["Ruffian"] = 3401388520] = "Ruffian";
    VehicleHash[VehicleHash["Ruiner"] = 4067225593] = "Ruiner";
    VehicleHash[VehicleHash["Ruiner2"] = 941494461] = "Ruiner2";
    VehicleHash[VehicleHash["Ruiner3"] = 777714999] = "Ruiner3";
    VehicleHash[VehicleHash["Rumpo"] = 1162065741] = "Rumpo";
    VehicleHash[VehicleHash["Rumpo2"] = 2518351607] = "Rumpo2";
    VehicleHash[VehicleHash["Rumpo3"] = 1475773103] = "Rumpo3";
    VehicleHash[VehicleHash["Ruston"] = 719660200] = "Ruston";
    VehicleHash[VehicleHash["SabreGT"] = 2609945748] = "SabreGT";
    VehicleHash[VehicleHash["SabreGT2"] = 223258115] = "SabreGT2";
    VehicleHash[VehicleHash["Sadler"] = 3695398481] = "Sadler";
    VehicleHash[VehicleHash["Sadler2"] = 734217681] = "Sadler2";
    VehicleHash[VehicleHash["Sanchez"] = 788045382] = "Sanchez";
    VehicleHash[VehicleHash["Sanchez2"] = 2841686334] = "Sanchez2";
    VehicleHash[VehicleHash["Sanctus"] = 1491277511] = "Sanctus";
    VehicleHash[VehicleHash["Sandking"] = 3105951696] = "Sandking";
    VehicleHash[VehicleHash["Sandking2"] = 989381445] = "Sandking2";
    VehicleHash[VehicleHash["Savage"] = 4212341271] = "Savage";
    VehicleHash[VehicleHash["Schafter2"] = 3039514899] = "Schafter2";
    VehicleHash[VehicleHash["Schafter3"] = 2809443750] = "Schafter3";
    VehicleHash[VehicleHash["Schafter4"] = 1489967196] = "Schafter4";
    VehicleHash[VehicleHash["Schafter5"] = 3406724313] = "Schafter5";
    VehicleHash[VehicleHash["Schafter6"] = 1922255844] = "Schafter6";
    VehicleHash[VehicleHash["Schwarzer"] = 3548084598] = "Schwarzer";
    VehicleHash[VehicleHash["Scorcher"] = 4108429845] = "Scorcher";
    VehicleHash[VehicleHash["Scrap"] = 2594165727] = "Scrap";
    VehicleHash[VehicleHash["Seashark"] = 3264692260] = "Seashark";
    VehicleHash[VehicleHash["Seashark2"] = 3678636260] = "Seashark2";
    VehicleHash[VehicleHash["Seashark3"] = 3983945033] = "Seashark3";
    VehicleHash[VehicleHash["Seminole"] = 1221512915] = "Seminole";
    VehicleHash[VehicleHash["Sentinel"] = 1349725314] = "Sentinel";
    VehicleHash[VehicleHash["Sentinel2"] = 873639469] = "Sentinel2";
    VehicleHash[VehicleHash["Serrano"] = 1337041428] = "Serrano";
    VehicleHash[VehicleHash["Seven70"] = 2537130571] = "Seven70";
    VehicleHash[VehicleHash["Shamal"] = 3080461301] = "Shamal";
    VehicleHash[VehicleHash["Sheava"] = 819197656] = "Sheava";
    VehicleHash[VehicleHash["Sheriff"] = 2611638396] = "Sheriff";
    VehicleHash[VehicleHash["Sheriff2"] = 1922257928] = "Sheriff2";
    VehicleHash[VehicleHash["Shotaro"] = 3889340782] = "Shotaro";
    VehicleHash[VehicleHash["Skylift"] = 1044954915] = "Skylift";
    VehicleHash[VehicleHash["SlamVan"] = 729783779] = "SlamVan";
    VehicleHash[VehicleHash["SlamVan2"] = 833469436] = "SlamVan2";
    VehicleHash[VehicleHash["SlamVan3"] = 1119641113] = "SlamVan3";
    VehicleHash[VehicleHash["Sovereign"] = 743478836] = "Sovereign";
    VehicleHash[VehicleHash["Specter"] = 1886268224] = "Specter";
    VehicleHash[VehicleHash["Specter2"] = 1074745671] = "Specter2";
    VehicleHash[VehicleHash["Speeder"] = 231083307] = "Speeder";
    VehicleHash[VehicleHash["Speeder2"] = 437538602] = "Speeder2";
    VehicleHash[VehicleHash["Speedo"] = 3484649228] = "Speedo";
    VehicleHash[VehicleHash["Speedo2"] = 728614474] = "Speedo2";
    VehicleHash[VehicleHash["Squalo"] = 400514754] = "Squalo";
    VehicleHash[VehicleHash["Stalion"] = 1923400478] = "Stalion";
    VehicleHash[VehicleHash["Stalion2"] = 3893323758] = "Stalion2";
    VehicleHash[VehicleHash["Stanier"] = 2817386317] = "Stanier";
    VehicleHash[VehicleHash["Stinger"] = 1545842587] = "Stinger";
    VehicleHash[VehicleHash["StingerGT"] = 2196019706] = "StingerGT";
    VehicleHash[VehicleHash["Stockade"] = 1747439474] = "Stockade";
    VehicleHash[VehicleHash["Stockade3"] = 4080511798] = "Stockade3";
    VehicleHash[VehicleHash["Stratum"] = 1723137093] = "Stratum";
    VehicleHash[VehicleHash["Stretch"] = 2333339779] = "Stretch";
    VehicleHash[VehicleHash["Stunt"] = 2172210288] = "Stunt";
    VehicleHash[VehicleHash["Submersible"] = 771711535] = "Submersible";
    VehicleHash[VehicleHash["Submersible2"] = 3228633070] = "Submersible2";
    VehicleHash[VehicleHash["Sultan"] = 970598228] = "Sultan";
    VehicleHash[VehicleHash["SultanRS"] = 3999278268] = "SultanRS";
    VehicleHash[VehicleHash["Suntrap"] = 4012021193] = "Suntrap";
    VehicleHash[VehicleHash["Superd"] = 1123216662] = "Superd";
    VehicleHash[VehicleHash["Supervolito"] = 710198397] = "Supervolito";
    VehicleHash[VehicleHash["Supervolito2"] = 2623428164] = "Supervolito2";
    VehicleHash[VehicleHash["Surano"] = 384071873] = "Surano";
    VehicleHash[VehicleHash["Surfer"] = 699456151] = "Surfer";
    VehicleHash[VehicleHash["Surfer2"] = 2983726598] = "Surfer2";
    VehicleHash[VehicleHash["Surge"] = 2400073108] = "Surge";
    VehicleHash[VehicleHash["Swift2"] = 1075432268] = "Swift2";
    VehicleHash[VehicleHash["Swift"] = 3955379698] = "Swift";
    VehicleHash[VehicleHash["T20"] = 1663218586] = "T20";
    VehicleHash[VehicleHash["Taco"] = 1951180813] = "Taco";
    VehicleHash[VehicleHash["Tailgater"] = 3286105550] = "Tailgater";
    VehicleHash[VehicleHash["Tampa"] = 972671128] = "Tampa";
    VehicleHash[VehicleHash["Tampa2"] = 3223586949] = "Tampa2";
    VehicleHash[VehicleHash["Tampa3"] = 3084515313] = "Tampa3";
    VehicleHash[VehicleHash["Tanker"] = 3564062519] = "Tanker";
    VehicleHash[VehicleHash["Tanker2"] = 1956216962] = "Tanker2";
    VehicleHash[VehicleHash["TankerCar"] = 586013744] = "TankerCar";
    VehicleHash[VehicleHash["Taxi"] = 3338918751] = "Taxi";
    VehicleHash[VehicleHash["Technical"] = 2198148358] = "Technical";
    VehicleHash[VehicleHash["Technical2"] = 1180875963] = "Technical2";
    VehicleHash[VehicleHash["Technical3"] = 1356124575] = "Technical3";
    VehicleHash[VehicleHash["Tempesta"] = 272929391] = "Tempesta";
    VehicleHash[VehicleHash["Thrust"] = 1836027715] = "Thrust";
    VehicleHash[VehicleHash["TipTruck"] = 48339065] = "TipTruck";
    VehicleHash[VehicleHash["TipTruck2"] = 3347205726] = "TipTruck2";
    VehicleHash[VehicleHash["Titan"] = 1981688531] = "Titan";
    VehicleHash[VehicleHash["Torero"] = 1504306544] = "Torero";
    VehicleHash[VehicleHash["Tornado"] = 464687292] = "Tornado";
    VehicleHash[VehicleHash["Tornado2"] = 1531094468] = "Tornado2";
    VehicleHash[VehicleHash["Tornado3"] = 1762279763] = "Tornado3";
    VehicleHash[VehicleHash["Tornado4"] = 2261744861] = "Tornado4";
    VehicleHash[VehicleHash["Tornado5"] = 2497353967] = "Tornado5";
    VehicleHash[VehicleHash["Tornado6"] = 2736567667] = "Tornado6";
    VehicleHash[VehicleHash["Toro"] = 1070967343] = "Toro";
    VehicleHash[VehicleHash["Toro2"] = 908897389] = "Toro2";
    VehicleHash[VehicleHash["Tourbus"] = 1941029835] = "Tourbus";
    VehicleHash[VehicleHash["TowTruck"] = 2971866336] = "TowTruck";
    VehicleHash[VehicleHash["TowTruck2"] = 3852654278] = "TowTruck2";
    VehicleHash[VehicleHash["TR2"] = 2078290630] = "TR2";
    VehicleHash[VehicleHash["TR3"] = 1784254509] = "TR3";
    VehicleHash[VehicleHash["TR4"] = 2091594960] = "TR4";
    VehicleHash[VehicleHash["Tractor"] = 1641462412] = "Tractor";
    VehicleHash[VehicleHash["Tractor2"] = 2218488798] = "Tractor2";
    VehicleHash[VehicleHash["Tractor3"] = 1445631933] = "Tractor3";
    VehicleHash[VehicleHash["TrailerLogs"] = 2016027501] = "TrailerLogs";
    VehicleHash[VehicleHash["TrailerLarge"] = 1502869817] = "TrailerLarge";
    VehicleHash[VehicleHash["Trailers"] = 3417488910] = "Trailers";
    VehicleHash[VehicleHash["Trailers2"] = 2715434129] = "Trailers2";
    VehicleHash[VehicleHash["Trailers3"] = 2236089197] = "Trailers3";
    VehicleHash[VehicleHash["Trailers4"] = 3194418602] = "Trailers4";
    VehicleHash[VehicleHash["TrailerSmall"] = 712162987] = "TrailerSmall";
    VehicleHash[VehicleHash["TrailerSmall2"] = 2413121211] = "TrailerSmall2";
    VehicleHash[VehicleHash["Trash"] = 1917016601] = "Trash";
    VehicleHash[VehicleHash["Trash2"] = 3039269212] = "Trash2";
    VehicleHash[VehicleHash["TRFlat"] = 2942498482] = "TRFlat";
    VehicleHash[VehicleHash["TriBike"] = 1127861609] = "TriBike";
    VehicleHash[VehicleHash["TriBike2"] = 3061159916] = "TriBike2";
    VehicleHash[VehicleHash["TriBike3"] = 3894672200] = "TriBike3";
    VehicleHash[VehicleHash["TrophyTruck"] = 101905590] = "TrophyTruck";
    VehicleHash[VehicleHash["TrophyTruck2"] = 3631668194] = "TrophyTruck2";
    VehicleHash[VehicleHash["Tropic"] = 290013743] = "Tropic";
    VehicleHash[VehicleHash["Tropic2"] = 1448677353] = "Tropic2";
    VehicleHash[VehicleHash["Tropos"] = 1887331236] = "Tropos";
    VehicleHash[VehicleHash["Tug"] = 2194326579] = "Tug";
    VehicleHash[VehicleHash["Turismor"] = 408192225] = "Turismor";
    VehicleHash[VehicleHash["Turismo2"] = 3312836369] = "Turismo2";
    VehicleHash[VehicleHash["TVTrailer"] = 2524324030] = "TVTrailer";
    VehicleHash[VehicleHash["Tyrus"] = 2067820283] = "Tyrus";
    VehicleHash[VehicleHash["UtilliTruck"] = 516990260] = "UtilliTruck";
    VehicleHash[VehicleHash["UtilliTruck2"] = 887537515] = "UtilliTruck2";
    VehicleHash[VehicleHash["UtilliTruck3"] = 2132890591] = "UtilliTruck3";
    VehicleHash[VehicleHash["Vacca"] = 338562499] = "Vacca";
    VehicleHash[VehicleHash["Vader"] = 4154065143] = "Vader";
    VehicleHash[VehicleHash["Vagner"] = 1939284556] = "Vagner";
    VehicleHash[VehicleHash["Valkyrie"] = 2694714877] = "Valkyrie";
    VehicleHash[VehicleHash["Valkyrie2"] = 1543134283] = "Valkyrie2";
    VehicleHash[VehicleHash["Velum"] = 2621610858] = "Velum";
    VehicleHash[VehicleHash["Velum2"] = 1077420264] = "Velum2";
    VehicleHash[VehicleHash["Verlierer2"] = 1102544804] = "Verlierer2";
    VehicleHash[VehicleHash["Vestra"] = 1341619767] = "Vestra";
    VehicleHash[VehicleHash["Vigero"] = 3469130167] = "Vigero";
    VehicleHash[VehicleHash["Vindicator"] = 2941886209] = "Vindicator";
    VehicleHash[VehicleHash["Virgo"] = 3796912450] = "Virgo";
    VehicleHash[VehicleHash["Virgo2"] = 3395457658] = "Virgo2";
    VehicleHash[VehicleHash["Virgo3"] = 16646064] = "Virgo3";
    VehicleHash[VehicleHash["Volatus"] = 2449479409] = "Volatus";
    VehicleHash[VehicleHash["Voltic"] = 2672523198] = "Voltic";
    VehicleHash[VehicleHash["Voltic2"] = 989294410] = "Voltic2";
    VehicleHash[VehicleHash["Voodoo"] = 2006667053] = "Voodoo";
    VehicleHash[VehicleHash["Voodoo2"] = 523724515] = "Voodoo2";
    VehicleHash[VehicleHash["Vortex"] = 3685342204] = "Vortex";
    VehicleHash[VehicleHash["Warrener"] = 1373123368] = "Warrener";
    VehicleHash[VehicleHash["Washington"] = 1777363799] = "Washington";
    VehicleHash[VehicleHash["Wastelander"] = 2382949506] = "Wastelander";
    VehicleHash[VehicleHash["Windsor"] = 1581459400] = "Windsor";
    VehicleHash[VehicleHash["Windsor2"] = 2364918497] = "Windsor2";
    VehicleHash[VehicleHash["Wolfsbane"] = 3676349299] = "Wolfsbane";
    VehicleHash[VehicleHash["XA21"] = 917809321] = "XA21";
    VehicleHash[VehicleHash["XLS"] = 1203490606] = "XLS";
    VehicleHash[VehicleHash["XLS2"] = 3862958888] = "XLS2";
    VehicleHash[VehicleHash["Youga"] = 65402552] = "Youga";
    VehicleHash[VehicleHash["Youga2"] = 1026149675] = "Youga2";
    VehicleHash[VehicleHash["Zentorno"] = 2891838741] = "Zentorno";
    VehicleHash[VehicleHash["Zion"] = 3172678083] = "Zion";
    VehicleHash[VehicleHash["Zion2"] = 3101863448] = "Zion2";
    VehicleHash[VehicleHash["ZombieA"] = 3285698347] = "ZombieA";
    VehicleHash[VehicleHash["ZombieB"] = 3724934023] = "ZombieB";
    VehicleHash[VehicleHash["ZType"] = 758895617] = "ZType";
})(VehicleHash = exports.VehicleHash || (exports.VehicleHash = {}));
