// fivem-js/lib/hashes/WeaponHash.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AmmoType = exports.VehicleWeaponHash = exports.WeaponHash = void 0;
var WeaponHash;
(function (WeaponHash) {
    WeaponHash[WeaponHash["Knife"] = 2578778090] = "Knife";
    WeaponHash[WeaponHash["Nightstick"] = 1737195953] = "Nightstick";
    WeaponHash[WeaponHash["Hammer"] = 1317494643] = "Hammer";
    WeaponHash[WeaponHash["Bat"] = 2508868239] = "Bat";
    WeaponHash[WeaponHash["GolfClub"] = 1141786504] = "GolfClub";
    WeaponHash[WeaponHash["Crowbar"] = 2227010557] = "Crowbar";
    WeaponHash[WeaponHash["Bottle"] = 4192643659] = "Bottle";
    WeaponHash[WeaponHash["SwitchBlade"] = 3756226112] = "SwitchBlade";
    WeaponHash[WeaponHash["Pistol"] = 453432689] = "Pistol";
    WeaponHash[WeaponHash["CombatPistol"] = 1593441988] = "CombatPistol";
    WeaponHash[WeaponHash["APPistol"] = 584646201] = "APPistol";
    WeaponHash[WeaponHash["Pistol50"] = 2578377531] = "Pistol50";
    WeaponHash[WeaponHash["FlareGun"] = 1198879012] = "FlareGun";
    WeaponHash[WeaponHash["MarksmanPistol"] = 3696079510] = "MarksmanPistol";
    WeaponHash[WeaponHash["Revolver"] = 3249783761] = "Revolver";
    WeaponHash[WeaponHash["MicroSMG"] = 324215364] = "MicroSMG";
    WeaponHash[WeaponHash["SMG"] = 736523883] = "SMG";
    WeaponHash[WeaponHash["AssaultSMG"] = 4024951519] = "AssaultSMG";
    WeaponHash[WeaponHash["CombatPDW"] = 171789620] = "CombatPDW";
    WeaponHash[WeaponHash["AssaultRifle"] = 3220176749] = "AssaultRifle";
    WeaponHash[WeaponHash["CarbineRifle"] = 2210333304] = "CarbineRifle";
    WeaponHash[WeaponHash["AdvancedRifle"] = 2937143193] = "AdvancedRifle";
    WeaponHash[WeaponHash["CompactRifle"] = 1649403952] = "CompactRifle";
    WeaponHash[WeaponHash["MG"] = 2634544996] = "MG";
    WeaponHash[WeaponHash["CombatMG"] = 2144741730] = "CombatMG";
    WeaponHash[WeaponHash["PumpShotgun"] = 487013001] = "PumpShotgun";
    WeaponHash[WeaponHash["SawnOffShotgun"] = 2017895192] = "SawnOffShotgun";
    WeaponHash[WeaponHash["AssaultShotgun"] = 3800352039] = "AssaultShotgun";
    WeaponHash[WeaponHash["BullpupShotgun"] = 2640438543] = "BullpupShotgun";
    WeaponHash[WeaponHash["DoubleBarrelShotgun"] = 4019527611] = "DoubleBarrelShotgun";
    WeaponHash[WeaponHash["StunGun"] = 911657153] = "StunGun";
    WeaponHash[WeaponHash["SniperRifle"] = 100416529] = "SniperRifle";
    WeaponHash[WeaponHash["HeavySniper"] = 205991906] = "HeavySniper";
    WeaponHash[WeaponHash["GrenadeLauncher"] = 2726580491] = "GrenadeLauncher";
    WeaponHash[WeaponHash["GrenadeLauncherSmoke"] = 1305664598] = "GrenadeLauncherSmoke";
    WeaponHash[WeaponHash["RPG"] = 2982836145] = "RPG";
    WeaponHash[WeaponHash["Minigun"] = 1119849093] = "Minigun";
    WeaponHash[WeaponHash["Grenade"] = 2481070269] = "Grenade";
    WeaponHash[WeaponHash["StickyBomb"] = 741814745] = "StickyBomb";
    WeaponHash[WeaponHash["SmokeGrenade"] = 4256991824] = "SmokeGrenade";
    WeaponHash[WeaponHash["BZGas"] = 2694266206] = "BZGas";
    WeaponHash[WeaponHash["Molotov"] = 615608432] = "Molotov";
    WeaponHash[WeaponHash["FireExtinguisher"] = 101631238] = "FireExtinguisher";
    WeaponHash[WeaponHash["PetrolCan"] = 883325847] = "PetrolCan";
    WeaponHash[WeaponHash["SNSPistol"] = 3218215474] = "SNSPistol";
    WeaponHash[WeaponHash["SpecialCarbine"] = 3231910285] = "SpecialCarbine";
    WeaponHash[WeaponHash["HeavyPistol"] = 3523564046] = "HeavyPistol";
    WeaponHash[WeaponHash["BullpupRifle"] = 2132975508] = "BullpupRifle";
    WeaponHash[WeaponHash["HomingLauncher"] = 1672152130] = "HomingLauncher";
    WeaponHash[WeaponHash["ProximityMine"] = 2874559379] = "ProximityMine";
    WeaponHash[WeaponHash["Snowball"] = 126349499] = "Snowball";
    WeaponHash[WeaponHash["VintagePistol"] = 137902532] = "VintagePistol";
    WeaponHash[WeaponHash["Dagger"] = 2460120199] = "Dagger";
    WeaponHash[WeaponHash["Firework"] = 2138347493] = "Firework";
    WeaponHash[WeaponHash["Musket"] = 2828843422] = "Musket";
    WeaponHash[WeaponHash["MarksmanRifle"] = 3342088282] = "MarksmanRifle";
    WeaponHash[WeaponHash["HeavyShotgun"] = 984333226] = "HeavyShotgun";
    WeaponHash[WeaponHash["Gusenberg"] = 1627465347] = "Gusenberg";
    WeaponHash[WeaponHash["Hatchet"] = 4191993645] = "Hatchet";
    WeaponHash[WeaponHash["Railgun"] = 1834241177] = "Railgun";
    WeaponHash[WeaponHash["Unarmed"] = 2725352035] = "Unarmed";
    WeaponHash[WeaponHash["KnuckleDuster"] = 3638508604] = "KnuckleDuster";
    WeaponHash[WeaponHash["Machete"] = 3713923289] = "Machete";
    WeaponHash[WeaponHash["MachinePistol"] = 3675956304] = "MachinePistol";
    WeaponHash[WeaponHash["Flashlight"] = 2343591895] = "Flashlight";
    WeaponHash[WeaponHash["Ball"] = 600439132] = "Ball";
    WeaponHash[WeaponHash["Flare"] = 1233104067] = "Flare";
    WeaponHash[WeaponHash["NightVision"] = 2803906140] = "NightVision";
    WeaponHash[WeaponHash["Parachute"] = 4222310262] = "Parachute";
    WeaponHash[WeaponHash["SweeperShotgun"] = 317205821] = "SweeperShotgun";
    WeaponHash[WeaponHash["BattleAxe"] = 3441901897] = "BattleAxe";
    WeaponHash[WeaponHash["CompactGrenadeLauncher"] = 125959754] = "CompactGrenadeLauncher";
    WeaponHash[WeaponHash["MiniSMG"] = 3173288789] = "MiniSMG";
    WeaponHash[WeaponHash["PipeBomb"] = 3125143736] = "PipeBomb";
    WeaponHash[WeaponHash["PoolCue"] = 2484171525] = "PoolCue";
    WeaponHash[WeaponHash["Wrench"] = 419712736] = "Wrench";
    WeaponHash[WeaponHash["PistolMk2"] = 3219281620] = "PistolMk2";
    WeaponHash[WeaponHash["AssaultRifleMk2"] = 961495388] = "AssaultRifleMk2";
    WeaponHash[WeaponHash["CarbineRifleMk2"] = 4208062921] = "CarbineRifleMk2";
    WeaponHash[WeaponHash["CombatMGMk2"] = 3686625920] = "CombatMGMk2";
    WeaponHash[WeaponHash["HeavySniperMk2"] = 177293209] = "HeavySniperMk2";
    WeaponHash[WeaponHash["SMGMk2"] = 2024373456] = "SMGMk2";
})(WeaponHash = exports.WeaponHash || (exports.WeaponHash = {}));
var VehicleWeaponHash;
(function (VehicleWeaponHash) {
    VehicleWeaponHash[VehicleWeaponHash["Invalid"] = -1] = "Invalid";
    VehicleWeaponHash[VehicleWeaponHash["Tank"] = 1945616459] = "Tank";
    VehicleWeaponHash[VehicleWeaponHash["SpaceRocket"] = -123497569] = "SpaceRocket";
    VehicleWeaponHash[VehicleWeaponHash["PlaneRocket"] = -821520672] = "PlaneRocket";
    VehicleWeaponHash[VehicleWeaponHash["PlayerLaser"] = -268631733] = "PlayerLaser";
    VehicleWeaponHash[VehicleWeaponHash["PlayerBullet"] = 1259576109] = "PlayerBullet";
    VehicleWeaponHash[VehicleWeaponHash["PlayerBuzzard"] = 1186503822] = "PlayerBuzzard";
    VehicleWeaponHash[VehicleWeaponHash["PlayerHunter"] = -1625648674] = "PlayerHunter";
    VehicleWeaponHash[VehicleWeaponHash["PlayerLazer"] = -494786007] = "PlayerLazer";
    VehicleWeaponHash[VehicleWeaponHash["EnemyLaser"] = 1566990507] = "EnemyLaser";
    VehicleWeaponHash[VehicleWeaponHash["SearchLight"] = -844344963] = "SearchLight";
    VehicleWeaponHash[VehicleWeaponHash["Radar"] = -764006018] = "Radar";
})(VehicleWeaponHash = exports.VehicleWeaponHash || (exports.VehicleWeaponHash = {}));
var AmmoType;
(function (AmmoType) {
    AmmoType[AmmoType["Melee"] = 0] = "Melee";
    AmmoType[AmmoType["FireExtinguisher"] = 1359393852] = "FireExtinguisher";
    AmmoType[AmmoType["Flare"] = 1808594799] = "Flare";
    AmmoType[AmmoType["FlareGun"] = 1173416293] = "FlareGun";
    AmmoType[AmmoType["PetrolCan"] = 3395492001] = "PetrolCan";
    AmmoType[AmmoType["Shotgun"] = 2416459067] = "Shotgun";
    AmmoType[AmmoType["Pistol"] = 1950175060] = "Pistol";
    AmmoType[AmmoType["Ball"] = 4287981158] = "Ball";
    AmmoType[AmmoType["Snowball"] = 2182627693] = "Snowball";
    AmmoType[AmmoType["Sniper"] = 1285032059] = "Sniper";
    AmmoType[AmmoType["AssaultRifle"] = 218444191] = "AssaultRifle";
    AmmoType[AmmoType["SMG"] = 1820140472] = "SMG";
    AmmoType[AmmoType["Molotov"] = 1446246869] = "Molotov";
    AmmoType[AmmoType["StunGun"] = 2955849184] = "StunGun";
    AmmoType[AmmoType["MG"] = 1788949567] = "MG";
    AmmoType[AmmoType["GrenadeLauncher"] = 1003267566] = "GrenadeLauncher";
    AmmoType[AmmoType["RPG"] = 1742569970] = "RPG";
    AmmoType[AmmoType["Minigun"] = 2680539266] = "Minigun";
    AmmoType[AmmoType["Firework"] = 2938367503] = "Firework";
    AmmoType[AmmoType["Railgun"] = 2034517757] = "Railgun";
    AmmoType[AmmoType["HomingLauncher"] = 2568293933] = "HomingLauncher";
    AmmoType[AmmoType["Grenade"] = 1003688881] = "Grenade";
    AmmoType[AmmoType["StickyBomb"] = 1411692055] = "StickyBomb";
    AmmoType[AmmoType["ProximityMine"] = 2938243239] = "ProximityMine";
    AmmoType[AmmoType["PipeBomb"] = 357983224] = "PipeBomb";
    AmmoType[AmmoType["SmokeGrenade"] = 3859679398] = "SmokeGrenade";
    AmmoType[AmmoType["BZGas"] = 2608103076] = "BZGas";
})(AmmoType = exports.AmmoType || (exports.AmmoType = {}));
