// fivem-js/lib/utils/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Quaternion = exports.Size = exports.getRandomInt = exports.clamp = exports.Color = exports.Point = exports.uuidv4 = exports.PointF = exports.LiteEvent = exports.measureStringWidthNoConvert = exports.measureString = exports.stringToArray = exports.Vector3 = void 0;
var Vector3_1 = require("./Vector3");
Object.defineProperty(exports, "Vector3", { enumerable: true, get: function () { return Vector3_1.Vector3; } });
var String_1 = require("./String");
Object.defineProperty(exports, "stringToArray", { enumerable: true, get: function () { return String_1.stringToArray; } });
Object.defineProperty(exports, "measureString", { enumerable: true, get: function () { return String_1.measureString; } });
Object.defineProperty(exports, "measureStringWidthNoConvert", { enumerable: true, get: function () { return String_1.measureStringWidthNoConvert; } });
var LiteEvent_1 = require("./LiteEvent");
Object.defineProperty(exports, "LiteEvent", { enumerable: true, get: function () { return LiteEvent_1.LiteEvent; } });
var PointF_1 = require("./PointF");
Object.defineProperty(exports, "PointF", { enumerable: true, get: function () { return PointF_1.PointF; } });
var UUIDV4_1 = require("./UUIDV4");
Object.defineProperty(exports, "uuidv4", { enumerable: true, get: function () { return UUIDV4_1.uuidv4; } });
var Point_1 = require("./Point");
Object.defineProperty(exports, "Point", { enumerable: true, get: function () { return Point_1.Point; } });
var Color_1 = require("./Color");
Object.defineProperty(exports, "Color", { enumerable: true, get: function () { return Color_1.Color; } });
var Math_1 = require("./Math");
Object.defineProperty(exports, "clamp", { enumerable: true, get: function () { return Math_1.clamp; } });
Object.defineProperty(exports, "getRandomInt", { enumerable: true, get: function () { return Math_1.getRandomInt; } });
var Size_1 = require("./Size");
Object.defineProperty(exports, "Size", { enumerable: true, get: function () { return Size_1.Size; } });
var Quaternion_1 = require("./Quaternion");
Object.defineProperty(exports, "Quaternion", { enumerable: true, get: function () { return Quaternion_1.Quaternion; } });
