// fivem-js/lib/ui/menu/items/panels/AbstractUIMenuPanel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AbstractUIMenuPanel = void 0;
const utils_1 = require("../../../../utils");
const __1 = require("../../../");
const __2 = require("../../");
class AbstractUIMenuPanel {
    constructor() {
        this.id = utils_1.uuidv4();
        this.enabled = true;
    }
    get ParentMenu() {
        return this.parentItem.parent;
    }
    get ParentItem() {
        return this.parentItem;
    }
    set ParentItem(value) {
        this.parentItem = value;
    }
    get Enabled() {
        return this.enabled;
    }
    set Enabled(value) {
        this.enabled = value;
    }
    get Height() {
        return this.background.size.height;
    }
    setVerticalPosition(y) {
        this.background.pos.Y = y;
    }
    draw() {
        this.background.size.width = 431 + this.ParentMenu.WidthOffset;
        this.background.pos.X = this.parentItem.offset.X;
        if (this.background instanceof __1.Sprite) {
            this.background.draw(__2.Menu.screenResolution);
        }
        else {
            this.background.draw(undefined, __2.Menu.screenResolution);
        }
    }
}
exports.AbstractUIMenuPanel = AbstractUIMenuPanel;
