// fivem-js/lib/enums/BlipSprite.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var BlipSprite;
(function (BlipSprite) {
    BlipSprite[BlipSprite["Standard"] = 1] = "Standard";
    BlipSprite[BlipSprite["BigBlip"] = 2] = "BigBlip";
    BlipSprite[BlipSprite["PoliceOfficer"] = 3] = "PoliceOfficer";
    BlipSprite[BlipSprite["PoliceArea"] = 4] = "PoliceArea";
    BlipSprite[BlipSprite["Square"] = 5] = "Square";
    BlipSprite[BlipSprite["Player"] = 6] = "Player";
    BlipSprite[BlipSprite["North"] = 7] = "North";
    BlipSprite[BlipSprite["Waypoint"] = 8] = "Waypoint";
    BlipSprite[BlipSprite["BigCircle"] = 9] = "BigCircle";
    BlipSprite[BlipSprite["BigCircleOutline"] = 10] = "BigCircleOutline";
    BlipSprite[BlipSprite["ArrowUpOutlined"] = 11] = "ArrowUpOutlined";
    BlipSprite[BlipSprite["ArrowDownOutlined"] = 12] = "ArrowDownOutlined";
    BlipSprite[BlipSprite["ArrowUp"] = 13] = "ArrowUp";
    BlipSprite[BlipSprite["ArrowDown"] = 14] = "ArrowDown";
    BlipSprite[BlipSprite["PoliceHelicopterAnimated"] = 15] = "PoliceHelicopterAnimated";
    BlipSprite[BlipSprite["Jet"] = 16] = "Jet";
    BlipSprite[BlipSprite["Number1"] = 17] = "Number1";
    BlipSprite[BlipSprite["Number2"] = 18] = "Number2";
    BlipSprite[BlipSprite["Number3"] = 19] = "Number3";
    BlipSprite[BlipSprite["Number4"] = 20] = "Number4";
    BlipSprite[BlipSprite["Number5"] = 21] = "Number5";
    BlipSprite[BlipSprite["Number6"] = 22] = "Number6";
    BlipSprite[BlipSprite["Number7"] = 23] = "Number7";
    BlipSprite[BlipSprite["Number8"] = 24] = "Number8";
    BlipSprite[BlipSprite["Number9"] = 25] = "Number9";
    BlipSprite[BlipSprite["Number10"] = 26] = "Number10";
    BlipSprite[BlipSprite["GTAOCrew"] = 27] = "GTAOCrew";
    BlipSprite[BlipSprite["GTAOFriendly"] = 28] = "GTAOFriendly";
    BlipSprite[BlipSprite["Lift"] = 36] = "Lift";
    BlipSprite[BlipSprite["RaceFinish"] = 38] = "RaceFinish";
    BlipSprite[BlipSprite["Safehouse"] = 40] = "Safehouse";
    BlipSprite[BlipSprite["PoliceOfficer2"] = 41] = "PoliceOfficer2";
    BlipSprite[BlipSprite["PoliceCarDot"] = 42] = "PoliceCarDot";
    BlipSprite[BlipSprite["PoliceHelicopter"] = 43] = "PoliceHelicopter";
    BlipSprite[BlipSprite["ChatBubble"] = 47] = "ChatBubble";
    BlipSprite[BlipSprite["Garage2"] = 50] = "Garage2";
    BlipSprite[BlipSprite["Drugs"] = 51] = "Drugs";
    BlipSprite[BlipSprite["Store"] = 52] = "Store";
    BlipSprite[BlipSprite["PoliceCar"] = 56] = "PoliceCar";
    BlipSprite[BlipSprite["PolicePlayer"] = 58] = "PolicePlayer";
    BlipSprite[BlipSprite["PoliceStation"] = 60] = "PoliceStation";
    BlipSprite[BlipSprite["Hospital"] = 61] = "Hospital";
    BlipSprite[BlipSprite["Helicopter"] = 64] = "Helicopter";
    BlipSprite[BlipSprite["StrangersAndFreaks"] = 65] = "StrangersAndFreaks";
    BlipSprite[BlipSprite["ArmoredTruck"] = 66] = "ArmoredTruck";
    BlipSprite[BlipSprite["TowTruck"] = 68] = "TowTruck";
    BlipSprite[BlipSprite["Barber"] = 71] = "Barber";
    BlipSprite[BlipSprite["LosSantosCustoms"] = 72] = "LosSantosCustoms";
    BlipSprite[BlipSprite["Clothes"] = 73] = "Clothes";
    BlipSprite[BlipSprite["TattooParlor"] = 75] = "TattooParlor";
    BlipSprite[BlipSprite["Simeon"] = 76] = "Simeon";
    BlipSprite[BlipSprite["Lester"] = 77] = "Lester";
    BlipSprite[BlipSprite["Michael"] = 78] = "Michael";
    BlipSprite[BlipSprite["Trevor"] = 79] = "Trevor";
    BlipSprite[BlipSprite["Rampage"] = 84] = "Rampage";
    BlipSprite[BlipSprite["VinewoodTours"] = 85] = "VinewoodTours";
    BlipSprite[BlipSprite["Lamar"] = 86] = "Lamar";
    BlipSprite[BlipSprite["Franklin"] = 88] = "Franklin";
    BlipSprite[BlipSprite["Chinese"] = 89] = "Chinese";
    BlipSprite[BlipSprite["Airport"] = 90] = "Airport";
    BlipSprite[BlipSprite["Bar"] = 93] = "Bar";
    BlipSprite[BlipSprite["BaseJump"] = 94] = "BaseJump";
    BlipSprite[BlipSprite["CarWash"] = 100] = "CarWash";
    BlipSprite[BlipSprite["ComedyClub"] = 102] = "ComedyClub";
    BlipSprite[BlipSprite["Dart"] = 103] = "Dart";
    BlipSprite[BlipSprite["FIB"] = 106] = "FIB";
    BlipSprite[BlipSprite["DollarSign"] = 108] = "DollarSign";
    BlipSprite[BlipSprite["Golf"] = 109] = "Golf";
    BlipSprite[BlipSprite["AmmuNation"] = 110] = "AmmuNation";
    BlipSprite[BlipSprite["Exile"] = 112] = "Exile";
    BlipSprite[BlipSprite["ShootingRange"] = 119] = "ShootingRange";
    BlipSprite[BlipSprite["Solomon"] = 120] = "Solomon";
    BlipSprite[BlipSprite["StripClub"] = 121] = "StripClub";
    BlipSprite[BlipSprite["Tennis"] = 122] = "Tennis";
    BlipSprite[BlipSprite["Triathlon"] = 126] = "Triathlon";
    BlipSprite[BlipSprite["OffRoadRaceFinish"] = 127] = "OffRoadRaceFinish";
    BlipSprite[BlipSprite["Key"] = 134] = "Key";
    BlipSprite[BlipSprite["MovieTheater"] = 135] = "MovieTheater";
    BlipSprite[BlipSprite["Music"] = 136] = "Music";
    BlipSprite[BlipSprite["Marijuana"] = 140] = "Marijuana";
    BlipSprite[BlipSprite["Hunting"] = 141] = "Hunting";
    BlipSprite[BlipSprite["ArmsTraffickingGround"] = 147] = "ArmsTraffickingGround";
    BlipSprite[BlipSprite["Nigel"] = 149] = "Nigel";
    BlipSprite[BlipSprite["AssaultRifle"] = 150] = "AssaultRifle";
    BlipSprite[BlipSprite["Bat"] = 151] = "Bat";
    BlipSprite[BlipSprite["Grenade"] = 152] = "Grenade";
    BlipSprite[BlipSprite["Health"] = 153] = "Health";
    BlipSprite[BlipSprite["Knife"] = 154] = "Knife";
    BlipSprite[BlipSprite["Molotov"] = 155] = "Molotov";
    BlipSprite[BlipSprite["Pistol"] = 156] = "Pistol";
    BlipSprite[BlipSprite["RPG"] = 157] = "RPG";
    BlipSprite[BlipSprite["Shotgun"] = 158] = "Shotgun";
    BlipSprite[BlipSprite["SMG"] = 159] = "SMG";
    BlipSprite[BlipSprite["Sniper"] = 160] = "Sniper";
    BlipSprite[BlipSprite["SonicWave"] = 161] = "SonicWave";
    BlipSprite[BlipSprite["PointOfInterest"] = 162] = "PointOfInterest";
    BlipSprite[BlipSprite["GTAOPassive"] = 163] = "GTAOPassive";
    BlipSprite[BlipSprite["GTAOUsingMenu"] = 164] = "GTAOUsingMenu";
    BlipSprite[BlipSprite["Link"] = 171] = "Link";
    BlipSprite[BlipSprite["Minigun"] = 173] = "Minigun";
    BlipSprite[BlipSprite["GrenadeLauncher"] = 174] = "GrenadeLauncher";
    BlipSprite[BlipSprite["Armor"] = 175] = "Armor";
    BlipSprite[BlipSprite["Castle"] = 176] = "Castle";
    BlipSprite[BlipSprite["Camera"] = 184] = "Camera";
    BlipSprite[BlipSprite["Handcuffs"] = 188] = "Handcuffs";
    BlipSprite[BlipSprite["Yoga"] = 197] = "Yoga";
    BlipSprite[BlipSprite["Cab"] = 198] = "Cab";
    BlipSprite[BlipSprite["Number11"] = 199] = "Number11";
    BlipSprite[BlipSprite["Number12"] = 200] = "Number12";
    BlipSprite[BlipSprite["Number13"] = 201] = "Number13";
    BlipSprite[BlipSprite["Number14"] = 202] = "Number14";
    BlipSprite[BlipSprite["Number15"] = 203] = "Number15";
    BlipSprite[BlipSprite["Number16"] = 204] = "Number16";
    BlipSprite[BlipSprite["Shrink"] = 205] = "Shrink";
    BlipSprite[BlipSprite["Epsilon"] = 206] = "Epsilon";
    BlipSprite[BlipSprite["PersonalVehicleCar"] = 225] = "PersonalVehicleCar";
    BlipSprite[BlipSprite["PersonalVehicleBike"] = 226] = "PersonalVehicleBike";
    BlipSprite[BlipSprite["Custody"] = 237] = "Custody";
    BlipSprite[BlipSprite["ArmsTraffickingAir"] = 251] = "ArmsTraffickingAir";
    BlipSprite[BlipSprite["Fairground"] = 266] = "Fairground";
    BlipSprite[BlipSprite["PropertyManagement"] = 267] = "PropertyManagement";
    BlipSprite[BlipSprite["Altruist"] = 269] = "Altruist";
    BlipSprite[BlipSprite["Enemy"] = 270] = "Enemy";
    BlipSprite[BlipSprite["Chop"] = 273] = "Chop";
    BlipSprite[BlipSprite["Dead"] = 274] = "Dead";
    BlipSprite[BlipSprite["Hooker"] = 279] = "Hooker";
    BlipSprite[BlipSprite["Friend"] = 280] = "Friend";
    BlipSprite[BlipSprite["BountyHit"] = 303] = "BountyHit";
    BlipSprite[BlipSprite["GTAOMission"] = 304] = "GTAOMission";
    BlipSprite[BlipSprite["GTAOSurvival"] = 305] = "GTAOSurvival";
    BlipSprite[BlipSprite["CrateDrop"] = 306] = "CrateDrop";
    BlipSprite[BlipSprite["PlaneDrop"] = 307] = "PlaneDrop";
    BlipSprite[BlipSprite["Sub"] = 308] = "Sub";
    BlipSprite[BlipSprite["Race"] = 309] = "Race";
    BlipSprite[BlipSprite["Deathmatch"] = 310] = "Deathmatch";
    BlipSprite[BlipSprite["ArmWrestling"] = 311] = "ArmWrestling";
    BlipSprite[BlipSprite["AmmuNationShootingRange"] = 313] = "AmmuNationShootingRange";
    BlipSprite[BlipSprite["RaceAir"] = 314] = "RaceAir";
    BlipSprite[BlipSprite["RaceCar"] = 315] = "RaceCar";
    BlipSprite[BlipSprite["RaceSea"] = 316] = "RaceSea";
    BlipSprite[BlipSprite["GarbageTruck"] = 318] = "GarbageTruck";
    BlipSprite[BlipSprite["SafehouseForSale"] = 350] = "SafehouseForSale";
    BlipSprite[BlipSprite["Package"] = 351] = "Package";
    BlipSprite[BlipSprite["MartinMadrazo"] = 352] = "MartinMadrazo";
    BlipSprite[BlipSprite["EnemyHelicopter"] = 353] = "EnemyHelicopter";
    BlipSprite[BlipSprite["Boost"] = 354] = "Boost";
    BlipSprite[BlipSprite["Devin"] = 355] = "Devin";
    BlipSprite[BlipSprite["Marina"] = 356] = "Marina";
    BlipSprite[BlipSprite["Garage"] = 357] = "Garage";
    BlipSprite[BlipSprite["GolfFlag"] = 358] = "GolfFlag";
    BlipSprite[BlipSprite["Hangar"] = 359] = "Hangar";
    BlipSprite[BlipSprite["Helipad"] = 360] = "Helipad";
    BlipSprite[BlipSprite["JerryCan"] = 361] = "JerryCan";
    BlipSprite[BlipSprite["Masks"] = 362] = "Masks";
    BlipSprite[BlipSprite["HeistSetup"] = 363] = "HeistSetup";
    BlipSprite[BlipSprite["Incapacitated"] = 364] = "Incapacitated";
    BlipSprite[BlipSprite["PickupSpawn"] = 365] = "PickupSpawn";
    BlipSprite[BlipSprite["BoilerSuit"] = 366] = "BoilerSuit";
    BlipSprite[BlipSprite["Completed"] = 367] = "Completed";
    BlipSprite[BlipSprite["Rockets"] = 368] = "Rockets";
    BlipSprite[BlipSprite["GarageForSale"] = 369] = "GarageForSale";
    BlipSprite[BlipSprite["HelipadForSale"] = 370] = "HelipadForSale";
    BlipSprite[BlipSprite["MarinaForSale"] = 371] = "MarinaForSale";
    BlipSprite[BlipSprite["HangarForSale"] = 372] = "HangarForSale";
    BlipSprite[BlipSprite["Business"] = 374] = "Business";
    BlipSprite[BlipSprite["BusinessForSale"] = 375] = "BusinessForSale";
    BlipSprite[BlipSprite["RaceBike"] = 376] = "RaceBike";
    BlipSprite[BlipSprite["Parachute"] = 377] = "Parachute";
    BlipSprite[BlipSprite["TeamDeathmatch"] = 378] = "TeamDeathmatch";
    BlipSprite[BlipSprite["RaceFoot"] = 379] = "RaceFoot";
    BlipSprite[BlipSprite["VehicleDeathmatch"] = 380] = "VehicleDeathmatch";
    BlipSprite[BlipSprite["Barry"] = 381] = "Barry";
    BlipSprite[BlipSprite["Dom"] = 382] = "Dom";
    BlipSprite[BlipSprite["MaryAnn"] = 383] = "MaryAnn";
    BlipSprite[BlipSprite["Cletus"] = 384] = "Cletus";
    BlipSprite[BlipSprite["Josh"] = 385] = "Josh";
    BlipSprite[BlipSprite["Minute"] = 386] = "Minute";
    BlipSprite[BlipSprite["Omega"] = 387] = "Omega";
    BlipSprite[BlipSprite["Tonya"] = 388] = "Tonya";
    BlipSprite[BlipSprite["Paparazzo"] = 389] = "Paparazzo";
    BlipSprite[BlipSprite["Crosshair"] = 390] = "Crosshair";
    BlipSprite[BlipSprite["Creator"] = 398] = "Creator";
    BlipSprite[BlipSprite["CreatorDirection"] = 399] = "CreatorDirection";
    BlipSprite[BlipSprite["Abigail"] = 400] = "Abigail";
    BlipSprite[BlipSprite["Blimp"] = 401] = "Blimp";
    BlipSprite[BlipSprite["Repair"] = 402] = "Repair";
    BlipSprite[BlipSprite["Testosterone"] = 403] = "Testosterone";
    BlipSprite[BlipSprite["Dinghy"] = 404] = "Dinghy";
    BlipSprite[BlipSprite["Fanatic"] = 405] = "Fanatic";
    BlipSprite[BlipSprite["Information"] = 407] = "Information";
    BlipSprite[BlipSprite["CaptureBriefcase"] = 408] = "CaptureBriefcase";
    BlipSprite[BlipSprite["LastTeamStanding"] = 409] = "LastTeamStanding";
    BlipSprite[BlipSprite["Boat"] = 410] = "Boat";
    BlipSprite[BlipSprite["CaptureHouse"] = 411] = "CaptureHouse";
    BlipSprite[BlipSprite["JerryCan2"] = 415] = "JerryCan2";
    BlipSprite[BlipSprite["RP"] = 416] = "RP";
    BlipSprite[BlipSprite["GTAOPlayerSafehouse"] = 417] = "GTAOPlayerSafehouse";
    BlipSprite[BlipSprite["GTAOPlayerSafehouseDead"] = 418] = "GTAOPlayerSafehouseDead";
    BlipSprite[BlipSprite["CaptureAmericanFlag"] = 419] = "CaptureAmericanFlag";
    BlipSprite[BlipSprite["CaptureFlag"] = 420] = "CaptureFlag";
    BlipSprite[BlipSprite["Tank"] = 421] = "Tank";
    BlipSprite[BlipSprite["HelicopterAnimated"] = 422] = "HelicopterAnimated";
    BlipSprite[BlipSprite["Plane"] = 423] = "Plane";
    BlipSprite[BlipSprite["PlayerNoColor"] = 425] = "PlayerNoColor";
    BlipSprite[BlipSprite["GunCar"] = 426] = "GunCar";
    BlipSprite[BlipSprite["Speedboat"] = 427] = "Speedboat";
    BlipSprite[BlipSprite["Heist"] = 428] = "Heist";
    BlipSprite[BlipSprite["Stopwatch"] = 430] = "Stopwatch";
    BlipSprite[BlipSprite["DollarSignCircled"] = 431] = "DollarSignCircled";
    BlipSprite[BlipSprite["Crosshair2"] = 432] = "Crosshair2";
    BlipSprite[BlipSprite["DollarSignSquared"] = 434] = "DollarSignSquared";
})(BlipSprite = exports.BlipSprite || (exports.BlipSprite = {}));
