// fivem-js/lib/enums/Bone.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Bone = void 0;
/* eslint-disable @typescript-eslint/naming-convention */
var Bone;
(function (Bone) {
    Bone[Bone["SKEL_ROOT"] = 0] = "SKEL_ROOT";
    Bone[Bone["SKEL_Pelvis"] = 11816] = "SKEL_Pelvis";
    Bone[Bone["SKEL_L_Thigh"] = 58271] = "SKEL_L_Thigh";
    Bone[Bone["SKEL_L_Calf"] = 63931] = "SKEL_L_Calf";
    Bone[Bone["SKEL_L_Foot"] = 14201] = "SKEL_L_Foot";
    Bone[Bone["SKEL_L_Toe0"] = 2108] = "SKEL_L_Toe0";
    Bone[Bone["IK_L_Foot"] = 65245] = "IK_L_Foot";
    Bone[Bone["PH_L_Foot"] = 57717] = "PH_L_Foot";
    Bone[Bone["MH_L_Knee"] = 46078] = "MH_L_Knee";
    Bone[Bone["SKEL_R_Thigh"] = 51826] = "SKEL_R_Thigh";
    Bone[Bone["SKEL_R_Calf"] = 36864] = "SKEL_R_Calf";
    Bone[Bone["SKEL_R_Foot"] = 52301] = "SKEL_R_Foot";
    Bone[Bone["SKEL_R_Toe0"] = 20781] = "SKEL_R_Toe0";
    Bone[Bone["IK_R_Foot"] = 35502] = "IK_R_Foot";
    Bone[Bone["PH_R_Foot"] = 24806] = "PH_R_Foot";
    Bone[Bone["MH_R_Knee"] = 16335] = "MH_R_Knee";
    Bone[Bone["RB_L_ThighRoll"] = 23639] = "RB_L_ThighRoll";
    Bone[Bone["RB_R_ThighRoll"] = 6442] = "RB_R_ThighRoll";
    Bone[Bone["SKEL_Spine_Root"] = 57597] = "SKEL_Spine_Root";
    Bone[Bone["SKEL_Spine0"] = 23553] = "SKEL_Spine0";
    Bone[Bone["SKEL_Spine1"] = 24816] = "SKEL_Spine1";
    Bone[Bone["SKEL_Spine2"] = 24817] = "SKEL_Spine2";
    Bone[Bone["SKEL_Spine3"] = 24818] = "SKEL_Spine3";
    Bone[Bone["SKEL_L_Clavicle"] = 64729] = "SKEL_L_Clavicle";
    Bone[Bone["SKEL_L_UpperArm"] = 45509] = "SKEL_L_UpperArm";
    Bone[Bone["SKEL_L_Forearm"] = 61163] = "SKEL_L_Forearm";
    Bone[Bone["SKEL_L_Hand"] = 18905] = "SKEL_L_Hand";
    Bone[Bone["SKEL_L_Finger00"] = 26610] = "SKEL_L_Finger00";
    Bone[Bone["SKEL_L_Finger01"] = 4089] = "SKEL_L_Finger01";
    Bone[Bone["SKEL_L_Finger02"] = 4090] = "SKEL_L_Finger02";
    Bone[Bone["SKEL_L_Finger10"] = 26611] = "SKEL_L_Finger10";
    Bone[Bone["SKEL_L_Finger11"] = 4169] = "SKEL_L_Finger11";
    Bone[Bone["SKEL_L_Finger12"] = 4170] = "SKEL_L_Finger12";
    Bone[Bone["SKEL_L_Finger20"] = 26612] = "SKEL_L_Finger20";
    Bone[Bone["SKEL_L_Finger21"] = 4185] = "SKEL_L_Finger21";
    Bone[Bone["SKEL_L_Finger22"] = 4186] = "SKEL_L_Finger22";
    Bone[Bone["SKEL_L_Finger30"] = 26613] = "SKEL_L_Finger30";
    Bone[Bone["SKEL_L_Finger31"] = 4137] = "SKEL_L_Finger31";
    Bone[Bone["SKEL_L_Finger32"] = 4138] = "SKEL_L_Finger32";
    Bone[Bone["SKEL_L_Finger40"] = 26614] = "SKEL_L_Finger40";
    Bone[Bone["SKEL_L_Finger41"] = 4153] = "SKEL_L_Finger41";
    Bone[Bone["SKEL_L_Finger42"] = 4154] = "SKEL_L_Finger42";
    Bone[Bone["PH_L_Hand"] = 60309] = "PH_L_Hand";
    Bone[Bone["IK_L_Hand"] = 36029] = "IK_L_Hand";
    Bone[Bone["RB_L_ForeArmRoll"] = 61007] = "RB_L_ForeArmRoll";
    Bone[Bone["RB_L_ArmRoll"] = 5232] = "RB_L_ArmRoll";
    Bone[Bone["MH_L_Elbow"] = 22711] = "MH_L_Elbow";
    Bone[Bone["SKEL_R_Clavicle"] = 10706] = "SKEL_R_Clavicle";
    Bone[Bone["SKEL_R_UpperArm"] = 40269] = "SKEL_R_UpperArm";
    Bone[Bone["SKEL_R_Forearm"] = 28252] = "SKEL_R_Forearm";
    Bone[Bone["SKEL_R_Hand"] = 57005] = "SKEL_R_Hand";
    Bone[Bone["SKEL_R_Finger00"] = 58866] = "SKEL_R_Finger00";
    Bone[Bone["SKEL_R_Finger01"] = 64016] = "SKEL_R_Finger01";
    Bone[Bone["SKEL_R_Finger02"] = 64017] = "SKEL_R_Finger02";
    Bone[Bone["SKEL_R_Finger10"] = 58867] = "SKEL_R_Finger10";
    Bone[Bone["SKEL_R_Finger11"] = 64096] = "SKEL_R_Finger11";
    Bone[Bone["SKEL_R_Finger12"] = 64097] = "SKEL_R_Finger12";
    Bone[Bone["SKEL_R_Finger20"] = 58868] = "SKEL_R_Finger20";
    Bone[Bone["SKEL_R_Finger21"] = 64112] = "SKEL_R_Finger21";
    Bone[Bone["SKEL_R_Finger22"] = 64113] = "SKEL_R_Finger22";
    Bone[Bone["SKEL_R_Finger30"] = 58869] = "SKEL_R_Finger30";
    Bone[Bone["SKEL_R_Finger31"] = 64064] = "SKEL_R_Finger31";
    Bone[Bone["SKEL_R_Finger32"] = 64065] = "SKEL_R_Finger32";
    Bone[Bone["SKEL_R_Finger40"] = 58870] = "SKEL_R_Finger40";
    Bone[Bone["SKEL_R_Finger41"] = 64080] = "SKEL_R_Finger41";
    Bone[Bone["SKEL_R_Finger42"] = 64081] = "SKEL_R_Finger42";
    Bone[Bone["PH_R_Hand"] = 28422] = "PH_R_Hand";
    Bone[Bone["IK_R_Hand"] = 6286] = "IK_R_Hand";
    Bone[Bone["RB_R_ForeArmRoll"] = 43810] = "RB_R_ForeArmRoll";
    Bone[Bone["RB_R_ArmRoll"] = 37119] = "RB_R_ArmRoll";
    Bone[Bone["MH_R_Elbow"] = 2992] = "MH_R_Elbow";
    Bone[Bone["SKEL_Neck_1"] = 39317] = "SKEL_Neck_1";
    Bone[Bone["SKEL_Head"] = 31086] = "SKEL_Head";
    Bone[Bone["IK_Head"] = 12844] = "IK_Head";
    Bone[Bone["FACIAL_facialRoot"] = 65068] = "FACIAL_facialRoot";
    Bone[Bone["FB_L_Brow_Out_000"] = 58331] = "FB_L_Brow_Out_000";
    Bone[Bone["FB_L_Lid_Upper_000"] = 45750] = "FB_L_Lid_Upper_000";
    Bone[Bone["FB_L_Eye_000"] = 25260] = "FB_L_Eye_000";
    Bone[Bone["FB_L_CheekBone_000"] = 21550] = "FB_L_CheekBone_000";
    Bone[Bone["FB_L_Lip_Corner_000"] = 29868] = "FB_L_Lip_Corner_000";
    Bone[Bone["FB_R_Lid_Upper_000"] = 43536] = "FB_R_Lid_Upper_000";
    Bone[Bone["FB_R_Eye_000"] = 27474] = "FB_R_Eye_000";
    Bone[Bone["FB_R_CheekBone_000"] = 19336] = "FB_R_CheekBone_000";
    Bone[Bone["FB_R_Brow_Out_000"] = 1356] = "FB_R_Brow_Out_000";
    Bone[Bone["FB_R_Lip_Corner_000"] = 11174] = "FB_R_Lip_Corner_000";
    Bone[Bone["FB_Brow_Centre_000"] = 37193] = "FB_Brow_Centre_000";
    Bone[Bone["FB_UpperLipRoot_000"] = 20178] = "FB_UpperLipRoot_000";
    Bone[Bone["FB_UpperLip_000"] = 61839] = "FB_UpperLip_000";
    Bone[Bone["FB_L_Lip_Top_000"] = 20279] = "FB_L_Lip_Top_000";
    Bone[Bone["FB_R_Lip_Top_000"] = 17719] = "FB_R_Lip_Top_000";
    Bone[Bone["FB_Jaw_000"] = 46240] = "FB_Jaw_000";
    Bone[Bone["FB_LowerLipRoot_000"] = 17188] = "FB_LowerLipRoot_000";
    Bone[Bone["FB_LowerLip_000"] = 20623] = "FB_LowerLip_000";
    Bone[Bone["FB_L_Lip_Bot_000"] = 47419] = "FB_L_Lip_Bot_000";
    Bone[Bone["FB_R_Lip_Bot_000"] = 49979] = "FB_R_Lip_Bot_000";
    Bone[Bone["FB_Tongue_000"] = 47495] = "FB_Tongue_000";
    Bone[Bone["RB_Neck_1"] = 35731] = "RB_Neck_1";
    Bone[Bone["IK_Root"] = 56604] = "IK_Root";
})(Bone = exports.Bone || (exports.Bone = {}));
