// fivem-js/lib/enums/Checkpoint.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckpointCustomIconStyle = exports.CheckpointIcon = void 0;
var CheckpointIcon;
(function (CheckpointIcon) {
    CheckpointIcon[CheckpointIcon["CylinderSingleArrow"] = 0] = "CylinderSingleArrow";
    CheckpointIcon[CheckpointIcon["CylinderDoubleArrow"] = 1] = "CylinderDoubleArrow";
    CheckpointIcon[CheckpointIcon["CylinderTripleArrow"] = 2] = "CylinderTripleArrow";
    CheckpointIcon[CheckpointIcon["CylinderCycleArrow"] = 3] = "CylinderCycleArrow";
    CheckpointIcon[CheckpointIcon["CylinderCheckerboard"] = 4] = "CylinderCheckerboard";
    CheckpointIcon[CheckpointIcon["CylinderSingleArrow2"] = 5] = "CylinderSingleArrow2";
    CheckpointIcon[CheckpointIcon["CylinderDoubleArrow2"] = 6] = "CylinderDoubleArrow2";
    CheckpointIcon[CheckpointIcon["CylinderTripleArrow2"] = 7] = "CylinderTripleArrow2";
    CheckpointIcon[CheckpointIcon["CylinderCycleArrow2"] = 8] = "CylinderCycleArrow2";
    CheckpointIcon[CheckpointIcon["CylinderCheckerboard2"] = 9] = "CylinderCheckerboard2";
    CheckpointIcon[CheckpointIcon["RingSingleArrow"] = 10] = "RingSingleArrow";
    CheckpointIcon[CheckpointIcon["RingDoubleArrow"] = 11] = "RingDoubleArrow";
    CheckpointIcon[CheckpointIcon["RingTripleArrow"] = 12] = "RingTripleArrow";
    CheckpointIcon[CheckpointIcon["RingCycleArrow"] = 13] = "RingCycleArrow";
    CheckpointIcon[CheckpointIcon["RingCheckerboard"] = 14] = "RingCheckerboard";
    CheckpointIcon[CheckpointIcon["SingleArrow"] = 15] = "SingleArrow";
    CheckpointIcon[CheckpointIcon["DoubleArrow"] = 16] = "DoubleArrow";
    CheckpointIcon[CheckpointIcon["TripleArrow"] = 17] = "TripleArrow";
    CheckpointIcon[CheckpointIcon["CycleArrow"] = 18] = "CycleArrow";
    CheckpointIcon[CheckpointIcon["Checkerboard"] = 19] = "Checkerboard";
    CheckpointIcon[CheckpointIcon["CylinderSingleArrow3"] = 20] = "CylinderSingleArrow3";
    CheckpointIcon[CheckpointIcon["CylinderDoubleArrow3"] = 21] = "CylinderDoubleArrow3";
    CheckpointIcon[CheckpointIcon["CylinderTripleArrow3"] = 22] = "CylinderTripleArrow3";
    CheckpointIcon[CheckpointIcon["CylinderCycleArrow3"] = 23] = "CylinderCycleArrow3";
    CheckpointIcon[CheckpointIcon["CylinderCheckerboard3"] = 24] = "CylinderCheckerboard3";
    CheckpointIcon[CheckpointIcon["CylinderSingleArrow4"] = 25] = "CylinderSingleArrow4";
    CheckpointIcon[CheckpointIcon["CylinderDoubleArrow4"] = 26] = "CylinderDoubleArrow4";
    CheckpointIcon[CheckpointIcon["CylinderTripleArrow4"] = 27] = "CylinderTripleArrow4";
    CheckpointIcon[CheckpointIcon["CylinderCycleArrow4"] = 28] = "CylinderCycleArrow4";
    CheckpointIcon[CheckpointIcon["CylinderCheckerboard4"] = 29] = "CylinderCheckerboard4";
    CheckpointIcon[CheckpointIcon["CylinderSingleArrow5"] = 30] = "CylinderSingleArrow5";
    CheckpointIcon[CheckpointIcon["CylinderDoubleArrow5"] = 31] = "CylinderDoubleArrow5";
    CheckpointIcon[CheckpointIcon["CylinderTripleArrow5"] = 32] = "CylinderTripleArrow5";
    CheckpointIcon[CheckpointIcon["CylinderCycleArrow5"] = 33] = "CylinderCycleArrow5";
    CheckpointIcon[CheckpointIcon["CylinderCheckerboard5"] = 34] = "CylinderCheckerboard5";
    CheckpointIcon[CheckpointIcon["RingPlaneUp"] = 35] = "RingPlaneUp";
    CheckpointIcon[CheckpointIcon["RingPlaneLeft"] = 36] = "RingPlaneLeft";
    CheckpointIcon[CheckpointIcon["RingPlaneRight"] = 37] = "RingPlaneRight";
    CheckpointIcon[CheckpointIcon["RingPlaneDown"] = 38] = "RingPlaneDown";
    CheckpointIcon[CheckpointIcon["Empty"] = 39] = "Empty";
    CheckpointIcon[CheckpointIcon["Ring"] = 40] = "Ring";
    CheckpointIcon[CheckpointIcon["Empty2"] = 41] = "Empty2";
    // CylinderCustomShape,
    // CylinderCustomShape2,
    // CylinderCustomShape3,
    CheckpointIcon[CheckpointIcon["Cyclinder"] = 45] = "Cyclinder";
    CheckpointIcon[CheckpointIcon["Cyclinder2"] = 46] = "Cyclinder2";
    CheckpointIcon[CheckpointIcon["Cyclinder3"] = 47] = "Cyclinder3";
})(CheckpointIcon = exports.CheckpointIcon || (exports.CheckpointIcon = {}));
var CheckpointCustomIconStyle;
(function (CheckpointCustomIconStyle) {
    CheckpointCustomIconStyle[CheckpointCustomIconStyle["Number"] = 0] = "Number";
    CheckpointCustomIconStyle[CheckpointCustomIconStyle["SingleArrow"] = 1] = "SingleArrow";
    CheckpointCustomIconStyle[CheckpointCustomIconStyle["DoubleArrow"] = 2] = "DoubleArrow";
    CheckpointCustomIconStyle[CheckpointCustomIconStyle["TripleArrow"] = 3] = "TripleArrow";
    CheckpointCustomIconStyle[CheckpointCustomIconStyle["Ring"] = 4] = "Ring";
    CheckpointCustomIconStyle[CheckpointCustomIconStyle["CycleArrow"] = 5] = "CycleArrow";
    CheckpointCustomIconStyle[CheckpointCustomIconStyle["Ring2"] = 6] = "Ring2";
    CheckpointCustomIconStyle[CheckpointCustomIconStyle["RingPointer"] = 7] = "RingPointer";
    CheckpointCustomIconStyle[CheckpointCustomIconStyle["SegmentedRing"] = 8] = "SegmentedRing";
    CheckpointCustomIconStyle[CheckpointCustomIconStyle["Sphere"] = 9] = "Sphere";
    CheckpointCustomIconStyle[CheckpointCustomIconStyle["Dollar"] = 10] = "Dollar";
    CheckpointCustomIconStyle[CheckpointCustomIconStyle["QuintupleLines"] = 11] = "QuintupleLines";
    CheckpointCustomIconStyle[CheckpointCustomIconStyle["BeastIcon"] = 12] = "BeastIcon";
})(CheckpointCustomIconStyle = exports.CheckpointCustomIconStyle || (exports.CheckpointCustomIconStyle = {}));
