// fivem-js/lib/enums/CursorSprite.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CursorSprite = void 0;
var CursorSprite;
(function (CursorSprite) {
    CursorSprite[CursorSprite["Normal"] = 1] = "Normal";
    CursorSprite[CursorSprite["LightArrow"] = 2] = "LightArrow";
    CursorSprite[CursorSprite["OpenHand"] = 3] = "OpenHand";
    CursorSprite[CursorSprite["GrabHand"] = 4] = "GrabHand";
    CursorSprite[CursorSprite["MiddleFinger"] = 5] = "MiddleFinger";
    CursorSprite[CursorSprite["LeftArrow"] = 6] = "LeftArrow";
    CursorSprite[CursorSprite["RightArrow"] = 7] = "RightArrow";
    CursorSprite[CursorSprite["UpArrow"] = 8] = "UpArrow";
    CursorSprite[CursorSprite["DownArrow"] = 9] = "DownArrow";
    CursorSprite[CursorSprite["HorizontalDoubleArrow"] = 10] = "HorizontalDoubleArrow";
    CursorSprite[CursorSprite["NormalWithPlus"] = 11] = "NormalWithPlus";
    CursorSprite[CursorSprite["NormalWithMinus"] = 12] = "NormalWithMinus";
})(CursorSprite = exports.CursorSprite || (exports.CursorSprite = {}));
