// fivem-js/lib/enums/ExplosionType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExplosionType = void 0;
/**
 * List of explosion sources.
 */
var ExplosionType;
(function (ExplosionType) {
    ExplosionType[ExplosionType["Grenade"] = 0] = "Grenade";
    ExplosionType[ExplosionType["GrenadeL"] = 1] = "GrenadeL";
    ExplosionType[ExplosionType["StickyBomb"] = 2] = "StickyBomb";
    ExplosionType[ExplosionType["Molotov1"] = 3] = "Molotov1";
    ExplosionType[ExplosionType["Rocket"] = 4] = "Rocket";
    ExplosionType[ExplosionType["TankShell"] = 5] = "TankShell";
    ExplosionType[ExplosionType["HiOctane"] = 6] = "HiOctane";
    ExplosionType[ExplosionType["Car"] = 7] = "Car";
    ExplosionType[ExplosionType["Plane"] = 8] = "Plane";
    ExplosionType[ExplosionType["PetrolPump"] = 9] = "PetrolPump";
    ExplosionType[ExplosionType["Bike"] = 10] = "Bike";
    ExplosionType[ExplosionType["Steam"] = 11] = "Steam";
    ExplosionType[ExplosionType["Flame"] = 12] = "Flame";
    ExplosionType[ExplosionType["WaterHydrant"] = 13] = "WaterHydrant";
    ExplosionType[ExplosionType["GasCanister"] = 14] = "GasCanister";
    ExplosionType[ExplosionType["Boat"] = 15] = "Boat";
    ExplosionType[ExplosionType["ShipDestroy"] = 16] = "ShipDestroy";
    ExplosionType[ExplosionType["Truck"] = 17] = "Truck";
    ExplosionType[ExplosionType["Bullet"] = 18] = "Bullet";
    ExplosionType[ExplosionType["SmokeGL"] = 19] = "SmokeGL";
    ExplosionType[ExplosionType["SmokeG"] = 20] = "SmokeG";
    ExplosionType[ExplosionType["BZGas"] = 21] = "BZGas";
    ExplosionType[ExplosionType["Flare"] = 22] = "Flare";
    ExplosionType[ExplosionType["GasCanister2"] = 23] = "GasCanister2";
    ExplosionType[ExplosionType["Extinguisher"] = 24] = "Extinguisher";
    ExplosionType[ExplosionType["ProgramAR"] = 25] = "ProgramAR";
    ExplosionType[ExplosionType["Train"] = 26] = "Train";
    ExplosionType[ExplosionType["Barrel"] = 27] = "Barrel";
    ExplosionType[ExplosionType["Propane"] = 28] = "Propane";
    ExplosionType[ExplosionType["Blimp"] = 29] = "Blimp";
    ExplosionType[ExplosionType["FlameExplode"] = 30] = "FlameExplode";
    ExplosionType[ExplosionType["Tanker"] = 31] = "Tanker";
    ExplosionType[ExplosionType["PlaneRocket"] = 32] = "PlaneRocket";
    ExplosionType[ExplosionType["VehicleBullet"] = 33] = "VehicleBullet";
    ExplosionType[ExplosionType["GasTank"] = 34] = "GasTank";
    ExplosionType[ExplosionType["FireWork"] = 35] = "FireWork";
    ExplosionType[ExplosionType["SnowBall"] = 36] = "SnowBall";
    ExplosionType[ExplosionType["ProxMine"] = 37] = "ProxMine";
    ExplosionType[ExplosionType["Valkyrie"] = 38] = "Valkyrie";
})(ExplosionType = exports.ExplosionType || (exports.ExplosionType = {}));
