// fivem-js/lib/models/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VehicleWindowCollection = exports.VehicleWindow = exports.VehicleWheelCollection = exports.VehicleWheel = exports.VehicleModCollection = exports.VehicleToggleMod = exports.VehicleMod = exports.VehicleDoorCollection = exports.VehicleDoor = exports.Vehicle = exports.Prop = exports.Player = exports.PedBoneCollection = exports.PedBone = exports.Ped = exports.EntityBoneCollection = exports.EntityBone = exports.Entity = void 0;
var Entity_1 = require("./Entity");
Object.defineProperty(exports, "Entity", { enumerable: true, get: function () { return Entity_1.Entity; } });
var EntityBone_1 = require("./EntityBone");
Object.defineProperty(exports, "EntityBone", { enumerable: true, get: function () { return EntityBone_1.EntityBone; } });
var EntityBoneCollection_1 = require("./EntityBoneCollection");
Object.defineProperty(exports, "EntityBoneCollection", { enumerable: true, get: function () { return EntityBoneCollection_1.EntityBoneCollection; } });
var Ped_1 = require("./Ped");
Object.defineProperty(exports, "Ped", { enumerable: true, get: function () { return Ped_1.Ped; } });
var PedBone_1 = require("./PedBone");
Object.defineProperty(exports, "PedBone", { enumerable: true, get: function () { return PedBone_1.PedBone; } });
var PedBoneCollection_1 = require("./PedBoneCollection");
Object.defineProperty(exports, "PedBoneCollection", { enumerable: true, get: function () { return PedBoneCollection_1.PedBoneCollection; } });
var Player_1 = require("./Player");
Object.defineProperty(exports, "Player", { enumerable: true, get: function () { return Player_1.Player; } });
var Prop_1 = require("./Prop");
Object.defineProperty(exports, "Prop", { enumerable: true, get: function () { return Prop_1.Prop; } });
var Vehicle_1 = require("./Vehicle");
Object.defineProperty(exports, "Vehicle", { enumerable: true, get: function () { return Vehicle_1.Vehicle; } });
var VehicleDoor_1 = require("./VehicleDoor");
Object.defineProperty(exports, "VehicleDoor", { enumerable: true, get: function () { return VehicleDoor_1.VehicleDoor; } });
var VehicleDoorCollection_1 = require("./VehicleDoorCollection");
Object.defineProperty(exports, "VehicleDoorCollection", { enumerable: true, get: function () { return VehicleDoorCollection_1.VehicleDoorCollection; } });
var VehicleMod_1 = require("./VehicleMod");
Object.defineProperty(exports, "VehicleMod", { enumerable: true, get: function () { return VehicleMod_1.VehicleMod; } });
var VehicleToggleMod_1 = require("./VehicleToggleMod");
Object.defineProperty(exports, "VehicleToggleMod", { enumerable: true, get: function () { return VehicleToggleMod_1.VehicleToggleMod; } });
var VehicleModCollection_1 = require("./VehicleModCollection");
Object.defineProperty(exports, "VehicleModCollection", { enumerable: true, get: function () { return VehicleModCollection_1.VehicleModCollection; } });
var VehicleWheel_1 = require("./VehicleWheel");
Object.defineProperty(exports, "VehicleWheel", { enumerable: true, get: function () { return VehicleWheel_1.VehicleWheel; } });
var VehicleWheelCollection_1 = require("./VehicleWheelCollection");
Object.defineProperty(exports, "VehicleWheelCollection", { enumerable: true, get: function () { return VehicleWheelCollection_1.VehicleWheelCollection; } });
var VehicleWindow_1 = require("./VehicleWindow");
Object.defineProperty(exports, "VehicleWindow", { enumerable: true, get: function () { return VehicleWindow_1.VehicleWindow; } });
var VehicleWindowCollection_1 = require("./VehicleWindowCollection");
Object.defineProperty(exports, "VehicleWindowCollection", { enumerable: true, get: function () { return VehicleWindowCollection_1.VehicleWindowCollection; } });
