// fivem-js/lib/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RelationshipGroup = exports.RaycastResult = exports.ParticleEffectAsset = exports.ParticleEffect = exports.GameplayCamera = exports.Checkpoint = exports.Camera = exports.Blip = exports.Audio = exports.Model = exports.World = exports.Game = void 0;
var Game_1 = require("./Game");
Object.defineProperty(exports, "Game", { enumerable: true, get: function () { return Game_1.Game; } });
var World_1 = require("./World");
Object.defineProperty(exports, "World", { enumerable: true, get: function () { return World_1.World; } });
var Model_1 = require("./Model");
Object.defineProperty(exports, "Model", { enumerable: true, get: function () { return Model_1.Model; } });
var Audio_1 = require("./Audio");
Object.defineProperty(exports, "Audio", { enumerable: true, get: function () { return Audio_1.Audio; } });
var Blip_1 = require("./Blip");
Object.defineProperty(exports, "Blip", { enumerable: true, get: function () { return Blip_1.Blip; } });
var Camera_1 = require("./Camera");
Object.defineProperty(exports, "Camera", { enumerable: true, get: function () { return Camera_1.Camera; } });
var Checkpoint_1 = require("./Checkpoint");
Object.defineProperty(exports, "Checkpoint", { enumerable: true, get: function () { return Checkpoint_1.Checkpoint; } });
var GameplayCamera_1 = require("./GameplayCamera");
Object.defineProperty(exports, "GameplayCamera", { enumerable: true, get: function () { return GameplayCamera_1.GameplayCamera; } });
var ParticleEffect_1 = require("./ParticleEffect");
Object.defineProperty(exports, "ParticleEffect", { enumerable: true, get: function () { return ParticleEffect_1.ParticleEffect; } });
var ParticleEffectAsset_1 = require("./ParticleEffectAsset");
Object.defineProperty(exports, "ParticleEffectAsset", { enumerable: true, get: function () { return ParticleEffectAsset_1.ParticleEffectAsset; } });
var Raycast_1 = require("./Raycast");
Object.defineProperty(exports, "RaycastResult", { enumerable: true, get: function () { return Raycast_1.RaycastResult; } });
var RelationshipGroup_1 = require("./RelationshipGroup");
Object.defineProperty(exports, "RelationshipGroup", { enumerable: true, get: function () { return RelationshipGroup_1.RelationshipGroup; } });
// Lets export all from folders
__exportStar(require("./models"), exports);
__exportStar(require("./utils"), exports);
__exportStar(require("./enums"), exports);
__exportStar(require("./hashes"), exports);
__exportStar(require("./ui"), exports);
